--- glibc-2.20/sysdeps/unix/make-syscalls.sh.orig	2014-12-20 14:00:04.210261443 +0100
+++ glibc-2.20/sysdeps/unix/make-syscalls.sh	2014-12-20 14:01:10.059904705 +0100
@@ -276,27 +276,27 @@
     vdso_symbol="${vdso_syscall%@*}"
     vdso_symver="${vdso_syscall#*@}"
     vdso_symver=`echo "$vdso_symver" | sed 's/\./_/g'`
-    echo "\
+    echo -E "\
 \$(foreach p,\$(sysd-rules-targets),\$(objpfx)\$(patsubst %,\$p,$file).os): \\
 		\$(..)sysdeps/unix/make-syscalls.sh\
 	\$(make-target-directory)
-	(echo '#include <dl-vdso.h>'; \\
-	 echo 'extern void *${strong}_ifunc (void) __asm (\"${strong}\");'; \\
-	 echo 'void *'; \\
-	 echo '${strong}_ifunc (void)'; \\
-	 echo '{'; \\
-	 echo '  PREPARE_VERSION_KNOWN (symver, ${vdso_symver});'; \\
-	 echo '  return _dl_vdso_vsym (\"${vdso_symbol}\", &symver);'; \\
-	 echo '}'; \\
-	 echo 'asm (\".type ${strong}, %gnu_indirect_function\");'; \\"
+	(echo -E '#include <dl-vdso.h>'; \\
+	 echo -E 'extern void *${strong}_ifunc (void) __asm (\"${strong}\");'; \\
+	 echo -E 'void *'; \\
+	 echo -E '${strong}_ifunc (void)'; \\
+	 echo -E '{'; \\
+	 echo -E '  PREPARE_VERSION_KNOWN (symver, ${vdso_symver});'; \\
+	 echo -E '  return _dl_vdso_vsym (\"${vdso_symbol}\", &symver);'; \\
+	 echo -E '}'; \\
+	 echo -E 'asm (\".type ${strong}, %gnu_indirect_function\");'; \\"
     # This is doing "libc_hidden_def (${strong})", but the compiler
     # doesn't know that we've defined ${strong} in the same file, so
     # we can't do it the normal way.
-    echo "\
-	 echo 'asm (\".globl __GI_${strong}\\n\"'; \\
-	 echo '     \"__GI_${strong} = ${strong}\");'; \\"
+    echo -E "\
+	 echo -E 'asm (\".globl __GI_${strong}\\n\"'; \\
+	 echo -E '     \"__GI_${strong} = ${strong}\");'; \\"
     emit_weak_aliases
-    echo '	) | $(compile-stdin.c) '"\
+    echo -E '	) | $(compile-stdin.c) '"\
 \$(foreach p,\$(patsubst %$file,%,\$(basename \$(@F))),\$(\$(p)CPPFLAGS))"
   fi
 
