--- malloc/Makefile~	Thu Jul 10 19:28:41 2003
+++ malloc/Makefile	Thu Jul 10 19:29:33 2003
@@ -86,7 +86,7 @@
 # The directory specified by --with-headers usually contains only the basic
 # kernel interface headers, not something like libgd.  So the simplest thing
 # is to presume that the standard system headers will be ok for this file.
-$(objpfx)memusagestat.o: sysincludes = # nothing
+$(objpfx)memusagestat.o: sysincludes = -I/usr/src/linux/include -I/usr/include
 endif
 endif
 
