diff -urN glibc-2.3.6.orig/sysdeps/unix/sysv/linux/dl-execstack.c glibc-2.3.6/sysdeps/unix/sysv/linux/dl-execstack.c
--- glibc-2.3.6.orig/sysdeps/unix/sysv/linux/dl-execstack.c	2005-07-18 06:11:19.000000000 +0200
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/dl-execstack.c	2005-12-10 10:14:37.000000000 +0100
@@ -62,6 +62,7 @@
 	no_growsupdown = true;
       else
 # endif
+      if (errno != EACCES)	/* PaX */
 	{
 	  result = errno;
 	  goto out;
@@ -89,7 +90,7 @@
 	page -= size;
       else
 	{
-	  if (errno != ENOMEM)	/* Unexpected failure mode.  */
+	  if (errno != ENOMEM && errno != EACCES)	/* Unexpected failure mode.  */	/* PaX */
 	    {
 	      result = errno;
 	      goto out;
@@ -115,7 +116,7 @@
 	page += size;
       else
 	{
-	  if (errno != ENOMEM)	/* Unexpected failure mode.  */
+	  if (errno != ENOMEM && errno != EACCES)	/* Unexpected failure mode.  */	/* PaX */
 	    {
 	      result = errno;
 	      goto out;
