--- glibc-2.5/sysdeps/powerpc/fpu/e_sqrt.c~	2006-10-06 18:33:08.000000000 +0200
+++ glibc-2.5/sysdeps/powerpc/fpu/e_sqrt.c	2006-10-06 18:32:25.000000000 +0200
@@ -159,6 +159,14 @@
 }
 
 
+#ifndef __CPU_HAS_FSQRT
+# if __WORDSIZE == 64 || defined _ARCH_PWR4
+#  define __CPU_HAS_FSQRT 1
+# else
+#  define __CPU_HAS_FSQRT ((GLRO(dl_hwcap) & PPC_FEATURE_64) != 0)
+# endif
+#endif
+
 #ifdef __STDC__
 double
 __ieee754_sqrt (double x)
--- glibc-2.5/sysdeps/powerpc/fpu/e_sqrtf.c~	2006-04-14 07:44:30.000000000 +0200
+++ glibc-2.5/sysdeps/powerpc/fpu/e_sqrtf.c	2006-10-06 18:32:45.000000000 +0200
@@ -135,6 +135,14 @@
 }
 
 
+#ifndef __CPU_HAS_FSQRT
+# if __WORDSIZE == 64 || defined _ARCH_PWR4
+#  define __CPU_HAS_FSQRT 1
+# else
+#  define __CPU_HAS_FSQRT ((GLRO(dl_hwcap) & PPC_FEATURE_64) != 0)
+# endif
+#endif
+
 #ifdef __STDC__
 float
 __ieee754_sqrtf (float x)
