--- libc.org/csu/Makefile.org	2004-10-14 22:01:32.382588324 +0200
+++ libc/csu/Makefile	2004-10-14 22:02:13.784718889 +0200
@@ -239,7 +239,7 @@
 	 esac; \
 	 files="$(all-Banner-files)";				\
 	 if test -n "$$files"; then				\
-	   echo "\"Available extensions:\\n\"";			\
+	   echo -e "\"Available extensions:\\\\n\"";			\
 	   sed -e '/^#/d' -e 's/^[[:space:]]*/	/'		\
 	       -e 's/^\(.*\)$$/\"\1\\n\"/' $$files;		\
 	 fi) > $@T
