diff --git a/sysdeps/unix/sysv/linux/powerpc/sys/user.h b/sysdeps/unix/sysv/linux/powerpc/sys/user.h
index e8a8aaa..1beafeb 100644
--- a/sysdeps/unix/sysv/linux/powerpc/sys/user.h
+++ b/sysdeps/unix/sysv/linux/powerpc/sys/user.h
@@ -21,6 +21,7 @@
 #define _SYS_USER_H	1
 #include <features.h>
 
+#include <asm/page.h>
 #include <asm/ptrace.h>
 
 struct user {
@@ -37,4 +38,10 @@ struct user {
 	char		u_comm[32];		/* user command name */
 };
 
+#define NBPG			PAGE_SIZE
+#define UPAGES			1
+#define HOST_TEXT_START_ADDR	(u.start_code)
+#define HOST_DATA_START_ADDR	(u.start_data)
+#define HOST_STACK_END_ADDR	(u.start_stack + u.u_ssize * NBPG)
+
 #endif  /* sys/user.h */
