--- glibc-2.5/sysdeps/powerpc/fpu/e_sqrt.c~	2006-10-06 18:33:08.000000000 +0200
+++ glibc-2.5/sysdeps/powerpc/fpu/e_sqrt.c	2006-10-06 18:32:25.000000000 +0200
@@ -159,6 +159,11 @@
 }
 
 
+#ifndef __CPU_HAS_FSQRT
+  /* If __CPU_HAS_FSQRT is not defined means we don't want optimization */
+# define __CPU_HAS_FSQRT 0
+#endif
+
 #ifdef __STDC__
 double
 __ieee754_sqrt (double x)
--- glibc-2.5/sysdeps/powerpc/fpu/e_sqrtf.c~	2006-04-14 07:44:30.000000000 +0200
+++ glibc-2.5/sysdeps/powerpc/fpu/e_sqrtf.c	2006-10-06 18:32:45.000000000 +0200
@@ -135,6 +135,11 @@
 }
 
 
+#ifndef __CPU_HAS_FSQRT
+  /* If __CPU_HAS_FSQRT is not defined means we don't want optimization */
+# define __CPU_HAS_FSQRT 0
+#endif
+
 #ifdef __STDC__
 float
 __ieee754_sqrtf (float x)
