--- glibc-2.3.6/locale/programs/ld-collate.c~	2005-11-08 17:12:52.000000000 +0000
+++ glibc-2.3.6/locale/programs/ld-collate.c	2005-11-08 17:12:52.000000000 +0000
@@ -2476,7 +2476,7 @@
       runp = runp->next;
     }
   /* Add 40% and find the next prime number.  */
-  elem_size = MIN (next_prime (elem_size * 1.4), 257);
+  elem_size = MIN (next_prime (elem_size * 1.4), 4097);
 
   /* Allocate the table.  Each entry consists of two words: the hash
      value and an index in a secondary table which provides the index
