diff -urNp -x '*.orig' glibc-2.33.org/manual/libc.texinfo glibc-2.33/manual/libc.texinfo
--- glibc-2.33.org/manual/libc.texinfo	2021-02-01 18:15:33.000000000 +0100
+++ glibc-2.33/manual/libc.texinfo	2021-03-20 14:27:49.037826328 +0100
@@ -14,9 +14,9 @@
 @include macros.texi
 
 @comment Tell install-info what to do.
-@dircategory Software libraries
+@dircategory Libraries:
 @direntry
-* Libc: (libc).                 C library.
+* Libc: (libc).				C library
 @end direntry
 @include dir-add.texi
 
