diff -dur glibc-2.3.3.orig/sysdeps/posix/getaddrinfo.c glibc-2.3.3/sysdeps/posix/getaddrinfo.c
--- glibc-2.3.3.orig/sysdeps/posix/getaddrinfo.c	2003-12-05 01:26:04.000000000 +0100
+++ glibc-2.3.3/sysdeps/posix/getaddrinfo.c	2004-02-02 12:13:23.982960898 +0100
@@ -1049,7 +1049,7 @@
 
       if (bits < 8)
 	{
-	  if ((*mask & (0xff >> bits)) == (*val & (0xff >> bits)))
+	  if ((*mask & (0xff00 >> bits)) == (*val & (0xff00 >> bits)))
 	    /* Match!  */
 	    break;
 	}
