diff -urN glibc-2.2.org/malloc/memusage.sh glibc-2.2/malloc/memusage.sh
--- glibc-2.2.org/malloc/memusage.sh	Wed Jan 10 11:45:39 2001
+++ glibc-2.2/malloc/memusage.sh	Wed Jan 10 11:45:51 2001
@@ -18,7 +18,7 @@
 # write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 # Boston, MA 02111-1307, USA.
 
-memusageso=@LIBDIR@/libmemusage.so
+memusageso=/usr/lib/libmemusage.so
 memusagestat=@BINDIR@/memusagestat
 
 # Print usage message.
