diff -Nur glibc-2.3.1/posix/Versions glibc-2.3.1.new/posix/Versions
--- glibc-2.3.1/posix/Versions	Thu Oct 10 07:24:52 2002
+++ glibc-2.3.1.new/posix/Versions	Thu Feb 20 21:24:05 2003
@@ -14,6 +14,9 @@
     # functions used by libstdc++ 2.7.2
     __waitpid;
 
+    # functions used by SunJava2-1.3.1/jre/lib/i386/libjava.so
+    __libc_wait; __libc_waitpid;
+
     # global variables
     __environ; _environ;
 
@@ -112,7 +115,7 @@
   GLIBC_PRIVATE {
     # functions which have an additional interface since they are
     # are cancelable.
-    __libc_wait; __libc_waitpid; __libc_pause; __libc_nanosleep; __libc_fork;
+    __libc_pause; __libc_nanosleep; __libc_fork;
     __libc_pread; __libc_pread64; __libc_pwrite; __libc_pwrite64;
     __waitid; __pselect;
   }
