--- glibc-2.3.3/sysdeps/unix/sysv/linux/alpha/rt_sigaction.S.orig	2003-09-14 16:32:30.000000000 +0000
+++ glibc-2.3.3/sysdeps/unix/sysv/linux/alpha/rt_sigaction.S	2004-01-13 09:00:10.000000000 +0000
@@ -45,7 +45,7 @@
 	ldah	a4, sigreturn(gp)		!gprelhigh
 	ldah	t1, rt_sigreturn(gp)		!gprelhigh
 	lda	a4, sigreturn(a4)		!gprellow
-	lda	t1, rt_sigreturn(a4)		!gprellow
+	lda	t1, rt_sigreturn(t1)		!gprellow
 	and	t0, 0x00000040, t0		# SA_SIGINFO
 	cmovne	t0, t1, a4
 0:	ldi	v0, __NR_rt_sigaction
