diff -urN glibc-2.3.3.org/csu/Makefile glibc-2.3.3/csu/Makefile
--- glibc-2.3.3.org/csu/Makefile	2004-01-01 22:13:10.105888239 +0100
+++ glibc-2.3.3/csu/Makefile	2004-01-01 22:13:38.533989059 +0100
@@ -232,12 +232,12 @@
 		     os=Linux; \
 		   fi; \
 		   echo "\"Compiled on a $$os $$version system" \
-			"on `date +%Y-%m-%d`.\\n\"" ;; \
+			"on `date +%Y-%m-%d`.\\\\n\"" ;; \
 	   *) ;; \
 	 esac; \
 	 files="$(all-Banner-files)";				\
 	 if test -n "$$files"; then				\
-	   echo "\"Available extensions:\\n\"";			\
+	   echo "\"Available extensions:\\\\n\"";			\
 	   sed -e '/^#/d' -e 's/^[[:space:]]*/	/'		\
 	       -e 's/^\(.*\)$$/\"\1\\n\"/' $$files;		\
 	 fi) > $@T
