--- glibc-2.2.2/elf/readlib.c~	Wed Mar 28 22:26:32 2001
+++ glibc-2.2.2/elf/readlib.c	Wed Mar 28 22:31:40 2001
@@ -87,8 +87,10 @@
   if (file == NULL)
     {
       /* No error for stale symlink.  */
-      if (is_link && strstr (file_name, ".so") != NULL)
+      if (is_link && strstr (file_name, ".so") != NULL) {
+	unlink (file_name);
 	return 1;
+      }
       error (0, 0, _("Input file %s not found.\n"), file_name);
       return 1;
     }
