diff -dur glibc-2.18.orig/localedata/SUPPORTED glibc-2.18/localedata/SUPPORTED
--- glibc-2.18.orig/localedata/SUPPORTED	2013-12-19 14:18:32.000000000 +0100
+++ glibc-2.18/localedata/SUPPORTED	2013-12-19 14:20:25.000000000 +0100
@@ -1,6 +1,7 @@
 # This file names the currently supported and somewhat tested locales.
 # If you have any additions please file a glibc bug report.
 SUPPORTED-LOCALES=\
+C.UTF-8/UTF-8 \
 aa_DJ.UTF-8/UTF-8 \
 aa_DJ/ISO-8859-1 \
 aa_ER/UTF-8 \
diff -dur glibc-2.18.orig/localedata/locales/C glibc-2.18/localedata/locales/C
--- glibc-2.18.orig/localedata/locales/C	2013-12-19 14:18:34.000000000 +0100
+++ glibc-2.18/localedata/locales/C	2013-12-19 14:47:06.000000000 +0100
@@ -8,18 +8,26 @@
 source     ""
 address    ""
 contact    ""
-email      "aurel32@debian.org"
+email      "pld-devel-en@lists.pld-linux.org"
 tel        ""
 fax        ""
 language   "C"
 territory  ""
 revision   "1.5"
-date       "2012-11-18"
+date       "2013-12-19"
 %
 category  "C:2012";LC_IDENTIFICATION
 category  "C:2012";LC_CTYPE
 category  "C:2012";LC_COLLATE
 category  "C:2012";LC_TIME
+category  "C:2013";LC_NUMERIC
+category  "C:2013";LC_MONETARY
+category  "C:2013";LC_MESSAGES
+category  "C:2013";LC_PAPER
+category  "C:2013";LC_NAME
+category  "C:2013";LC_ADDRESS
+category  "C:2013";LC_TELEPHONE
+category  "C:2013";LC_MEASUREMENT
 END LC_IDENTIFICATION
 
 LC_CTYPE
@@ -395036,3 +395044,70 @@
 <U0025><U005A><U0020><U0025><U0059>"
 
 END LC_TIME
+
+LC_NUMERIC
+copy	"POSIX"
+END LC_NUMERIC
+
+LC_MONETARY
+% This is the POSIX Locale definition for
+% the LC_MONETARY category.
+%
+%int_curr_symbol     ""
+currency_symbol     ""
+mon_decimal_point   "<U002E>"
+mon_thousands_sep   ""
+mon_grouping        -1
+positive_sign       ""
+negative_sign       ""
+int_frac_digits     -1
+frac_digits         -1
+p_cs_precedes       -1
+p_sep_by_space      -1
+n_cs_precedes       -1
+n_sep_by_space      -1
+p_sign_posn         -1
+n_sign_posn         -1
+%
+
+% value from POSIX gives: "value of field `int_curr_symbol' has wrong length"
+int_curr_symbol     "<U0055><U0053><U0044><U0020>"
+END LC_MONETARY
+
+LC_MESSAGES
+copy	"POSIX"
+END LC_MESSAGES
+
+LC_PAPER
+% from locale/C-paper.c
+height   297
+width    210
+END LC_PAPER
+
+LC_NAME
+% from locale/C-name.c
+name_fmt    "<U0025><U0070><U0025><U0074><U0025><U0067><U0025><U0074><U0025><U006D><U0025><U0074><U0025><U0066>"
+name_miss   ""
+name_mr     ""
+name_mrs    ""
+name_ms     ""
+END LC_NAME
+
+LC_ADDRESS
+% from locale/C-address.c
+postal_fmt    "<U0025><U0061><U0025><U004E><U0025><U0066><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0043><U002D><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+END LC_ADDRESS
+
+LC_TELEPHONE
+% from locale/C-telephone.c
+tel_int_fmt    "<U002B><U0025><U0063><U0020><U0025><U0061><U0020><U0025><U006C>"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+% from locale/C-measurement.c
+measurement    1
+END LC_MEASUREMENT
