--- libc/localedata/locales/zu_ZA~	Thu Nov 11 23:34:33 2004
+++ libc/localedata/locales/zu_ZA	Fri Nov 12 10:50:47 2004
@@ -56,10 +56,343 @@
 END LC_CTYPE
 
 LC_COLLATE
-% Copy the template from ISO/IEC 14651 i.e.
-% use the rules there when making ordered lists of words.
-%
 copy "iso14651_t1"
+
+%% a b bh c ch d dl e f g gc gq gx h hh hl i j k kh kl l m
+%% n nc ng nq nx ny o p ph q qh r s sh t th ts tsh u v w x
+%% xh y z
+
+collating-symbol  <bh>
+collating-element <B-H> from "<U0042><U0048>"
+collating-element <B-h> from "<U0042><U0068>"
+collating-element <b-H> from "<U0062><U0048>"
+collating-element <b-h> from "<U0062><U0068>"
+
+collating-symbol  <ch>
+collating-element <C-H> from "<U0043><U0048>"
+collating-element <C-h> from "<U0043><U0068>"
+collating-element <c-H> from "<U0063><U0048>"
+collating-element <c-h> from "<U0063><U0068>"
+
+collating-symbol  <dl>
+collating-element <D-L> from "<U0044><U004C>"
+collating-element <D-l> from "<U0044><U006C>"
+collating-element <d-L> from "<U0064><U004C>"
+collating-element <d-l> from "<U0064><U006C>"
+
+collating-symbol  <gc>
+collating-element <G-C> from "<U0047><U0043>"
+collating-element <G-c> from "<U0047><U0063>"
+collating-element <g-C> from "<U0067><U0043>"
+collating-element <g-c> from "<U0067><U0063>"
+
+collating-symbol  <gq>
+collating-element <G-Q> from "<U0047><U0051>"
+collating-element <G-q> from "<U0047><U0071>"
+collating-element <g-Q> from "<U0067><U0051>"
+collating-element <g-q> from "<U0067><U0071>"
+
+collating-symbol  <gx>
+collating-element <G-X> from "<U0047><U0058>"
+collating-element <G-x> from "<U0047><U0078>"
+collating-element <g-X> from "<U0067><U0058>"
+collating-element <g-x> from "<U0067><U0078>"
+
+collating-symbol  <hh>
+collating-element <H-H> from "<U0048><U0048>"
+collating-element <H-h> from "<U0048><U0068>"
+collating-element <h-H> from "<U0068><U0048>"
+collating-element <h-h> from "<U0068><U0068>"
+
+collating-symbol  <hl>
+collating-element <H-L> from "<U0048><U004C>"
+collating-element <H-l> from "<U0048><U006C>"
+collating-element <h-L> from "<U0068><U004C>"
+collating-element <h-l> from "<U0068><U006C>"
+
+collating-symbol  <kh>
+collating-element <K-H> from "<U004B><U0048>"
+collating-element <K-h> from "<U004B><U0068>"
+collating-element <k-H> from "<U006B><U0048>"
+collating-element <k-h> from "<U006B><U0068>"
+
+collating-symbol  <kl>
+collating-element <K-L> from "<U004B><U004C>"
+collating-element <K-l> from "<U004B><U006C>"
+collating-element <k-L> from "<U006B><U004C>"
+collating-element <k-l> from "<U006B><U006C>"
+
+collating-symbol  <nc>
+collating-element <N-C> from "<U004E><U0043>"
+collating-element <N-c> from "<U004E><U0063>"
+collating-element <n-C> from "<U006E><U0043>"
+collating-element <n-c> from "<U006E><U0063>"
+
+collating-symbol  <ng>
+collating-element <N-G> from "<U004E><U0047>"
+collating-element <N-g> from "<U004E><U0067>"
+collating-element <n-G> from "<U006E><U0047>"
+collating-element <n-g> from "<U006E><U0067>"
+
+collating-symbol  <nq>
+collating-element <N-Q> from "<U004E><U0051>"
+collating-element <N-q> from "<U004E><U0071>"
+collating-element <n-Q> from "<U006E><U0051>"
+collating-element <n-q> from "<U006E><U0071>"
+
+collating-symbol  <nx>
+collating-element <N-X> from "<U004E><U0058>"
+collating-element <N-x> from "<U004E><U0078>"
+collating-element <n-X> from "<U006E><U0058>"
+collating-element <n-x> from "<U006E><U0078>"
+
+collating-symbol  <ny>
+collating-element <N-Y> from "<U004E><U0059>"
+collating-element <N-y> from "<U004E><U0079>"
+collating-element <n-Y> from "<U006E><U0059>"
+collating-element <n-y> from "<U006E><U0079>"
+
+collating-symbol  <ph>
+collating-element <P-H> from "<U0050><U0048>"
+collating-element <P-h> from "<U0050><U0068>"
+collating-element <p-H> from "<U0070><U0048>"
+collating-element <p-h> from "<U0070><U0068>"
+
+collating-symbol  <qh>
+collating-element <Q-H> from "<U0051><U0048>"
+collating-element <Q-h> from "<U0051><U0068>"
+collating-element <q-H> from "<U0071><U0048>"
+collating-element <q-h> from "<U0071><U0068>"
+
+collating-symbol  <sh>
+collating-element <S-H> from "<U0053><U0048>"
+collating-element <S-h> from "<U0053><U0068>"
+collating-element <s-H> from "<U0073><U0048>"
+collating-element <s-h> from "<U0073><U0068>"
+
+% symbol <th> is already used in iso14651_t1 
+collating-symbol  <th0>
+collating-element <T-H> from "<U0054><U0048>"
+collating-element <T-h> from "<U0054><U0068>"
+collating-element <t-H> from "<U0074><U0048>"
+collating-element <t-h> from "<U0074><U0068>"
+
+collating-symbol  <ts>
+collating-element <T-S> from "<U0054><U0053>"
+collating-element <T-s> from "<U0054><U0073>"
+collating-element <t-S> from "<U0074><U0053>"
+collating-element <t-s> from "<U0074><U0073>"
+
+collating-symbol  <tsh>
+collating-element <T-S-H> from "<U0054><U0053><U0048>"
+collating-element <T-S-h> from "<U0054><U0053><U0068>"
+collating-element <T-s-H> from "<U0054><U0073><U0048>"
+collating-element <T-s-h> from "<U0054><U0073><U0068>"
+collating-element <t-S-H> from "<U0074><U0053><U0048>"
+collating-element <t-S-h> from "<U0074><U0053><U0068>"
+collating-element <t-s-H> from "<U0074><U0073><U0048>"
+collating-element <t-s-h> from "<U0074><U0073><U0068>"
+
+collating-symbol  <xh>
+collating-element <X-H> from "<U0058><U0048>"
+collating-element <X-h> from "<U0058><U0068>"
+collating-element <x-H> from "<U0078><U0048>"
+collating-element <x-h> from "<U0078><U0068>"
+
+collating-symbol <CAP-CAP>
+collating-symbol <CAP-MIN>
+collating-symbol <MIN-CAP>
+collating-symbol <MIN-MIN>
+collating-symbol <CAP-CAP-CAP>
+collating-symbol <CAP-CAP-MIN>
+collating-symbol <CAP-MIN-CAP>
+collating-symbol <CAP-MIN-MIN>
+collating-symbol <MIN-CAP-CAP>
+collating-symbol <MIN-CAP-MIN>
+collating-symbol <MIN-MIN-CAP>
+collating-symbol <MIN-MIN-MIN>
+
+reorder-after <MIN>
+<MIN-MIN>
+<MIN-CAP>
+<MIN-MIN-MIN>
+<MIN-MIN-CAP>
+<MIN-CAP-MIN>
+<MIN-CAP-CAP>
+
+reorder-after <CAP>
+<CAP-MIN>
+<CAP-CAP>
+<CAP-MIN-MIN>
+<CAP-MIN-CAP>
+<CAP-CAP-MIN>
+<CAP-CAP-CAP>
+
+reorder-after <BEFORE-C>
+<bh>
+reorder-after <BEFORE-D>
+<ch>
+reorder-after <BEFORE-E>
+<dl>
+reorder-after <BEFORE-H>
+<gc>
+<gq>
+<gx>
+reorder-after <BEFORE-I>
+<hh>
+<hl>
+reorder-after <BEFORE-L>
+<kh>
+<kl>
+reorder-after <BEFORE-O>
+<nc>
+<ng>
+<nq>
+<nx>
+<ny>
+reorder-after <BEFORE-Q>
+<ph>
+reorder-after <BEFORE-R>
+<qh>
+reorder-after <BEFORE-T>
+<sh>
+reorder-after <BEFORE-U>
+<th0>
+<ts>
+<tsh>
+reorder-after <BEFORE-Y>
+<xh>
+
+reorder-after <U0042>
+<B-h>		<bh>;<BASE>;<CAP-MIN>;IGNORE
+<B-H>		<bh>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0062>
+<b-h>		<bh>;<BASE>;<MIN-MIN>;IGNORE
+<b-H>		<bh>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0043>
+<C-h>		<ch>;<BASE>;<CAP-MIN>;IGNORE
+<C-H>		<ch>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0063>
+<c-h>		<ch>;<BASE>;<MIN-MIN>;IGNORE
+<c-H>		<ch>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0044>
+<D-l>		<dl>;<BASE>;<CAP-MIN>;IGNORE
+<D-L>		<dl>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0064>
+<d-l>		<dl>;<BASE>;<MIN-MIN>;IGNORE
+<d-L>		<dl>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0047>
+<G-c>		<gc>;<BASE>;<CAP-MIN>;IGNORE
+<G-C>		<gc>;<BASE>;<CAP-CAP>;IGNORE
+<G-q>		<gq>;<BASE>;<CAP-MIN>;IGNORE
+<G-Q>		<gq>;<BASE>;<CAP-CAP>;IGNORE
+<G-x>		<gx>;<BASE>;<CAP-MIN>;IGNORE
+<G-X>		<gx>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0067>
+<g-c>		<gc>;<BASE>;<MIN-MIN>;IGNORE
+<g-C>		<gc>;<BASE>;<MIN-CAP>;IGNORE
+<g-q>		<gq>;<BASE>;<MIN-MIN>;IGNORE
+<g-Q>		<gq>;<BASE>;<MIN-CAP>;IGNORE
+<g-x>		<gx>;<BASE>;<MIN-MIN>;IGNORE
+<g-X>		<gx>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0048>
+<H-h>		<hh>;<BASE>;<CAP-MIN>;IGNORE
+<H-H>		<hh>;<BASE>;<CAP-CAP>;IGNORE
+<H-l>		<hl>;<BASE>;<CAP-MIN>;IGNORE
+<H-L>		<hl>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0068>
+<h-h>		<hh>;<BASE>;<MIN-MIN>;IGNORE
+<h-H>		<hh>;<BASE>;<MIN-CAP>;IGNORE
+<h-l>		<hl>;<BASE>;<MIN-MIN>;IGNORE
+<h-L>		<hl>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U004B>
+<K-h>		<kh>;<BASE>;<CAP-MIN>;IGNORE
+<K-H>		<kh>;<BASE>;<CAP-CAP>;IGNORE
+<K-l>		<kl>;<BASE>;<CAP-MIN>;IGNORE
+<K-L>		<kl>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U006B>
+<k-h>		<kh>;<BASE>;<MIN-MIN>;IGNORE
+<k-H>		<kh>;<BASE>;<MIN-CAP>;IGNORE
+<k-l>		<kl>;<BASE>;<MIN-MIN>;IGNORE
+<k-L>		<kl>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U004E>
+<N-c>		<nc>;<BASE>;<CAP-MIN>;IGNORE
+<N-C>		<nc>;<BASE>;<CAP-CAP>;IGNORE
+<N-g>		<ng>;<BASE>;<CAP-MIN>;IGNORE
+<N-G>		<ng>;<BASE>;<CAP-CAP>;IGNORE
+<N-q>		<nq>;<BASE>;<CAP-MIN>;IGNORE
+<N-Q>		<nq>;<BASE>;<CAP-CAP>;IGNORE
+<N-x>		<nx>;<BASE>;<CAP-MIN>;IGNORE
+<N-X>		<nx>;<BASE>;<CAP-CAP>;IGNORE
+<N-y>		<ny>;<BASE>;<CAP-MIN>;IGNORE
+<N-Y>		<ny>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U006E>
+<n-c>		<nc>;<BASE>;<MIN-MIN>;IGNORE
+<n-C>		<nc>;<BASE>;<MIN-CAP>;IGNORE
+<n-g>		<ng>;<BASE>;<MIN-MIN>;IGNORE
+<n-G>		<ng>;<BASE>;<MIN-CAP>;IGNORE
+<n-q>		<nq>;<BASE>;<MIN-MIN>;IGNORE
+<n-Q>		<nq>;<BASE>;<MIN-CAP>;IGNORE
+<n-x>		<nx>;<BASE>;<MIN-MIN>;IGNORE
+<n-X>		<nx>;<BASE>;<MIN-CAP>;IGNORE
+<n-y>		<ny>;<BASE>;<MIN-MIN>;IGNORE
+<n-Y>		<ny>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0050>
+<P-h>		<ph>;<BASE>;<CAP-MIN>;IGNORE
+<P-H>		<ph>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0070>
+<p-h>		<ph>;<BASE>;<MIN-MIN>;IGNORE
+<p-H>		<ph>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0051>
+<Q-h>		<qh>;<BASE>;<CAP-MIN>;IGNORE
+<Q-H>		<qh>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0071>
+<q-h>		<qh>;<BASE>;<MIN-MIN>;IGNORE
+<q-H>		<qh>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0053>
+<S-h>		<sh>;<BASE>;<CAP-MIN>;IGNORE
+<S-H>		<sh>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0073>
+<s-h>		<sh>;<BASE>;<MIN-MIN>;IGNORE
+<s-H>		<sh>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0054>
+<T-h>		<th0>;<BASE>;<CAP-MIN>;IGNORE
+<T-H>		<th0>;<BASE>;<CAP-CAP>;IGNORE
+<T-s>		<ts>;<BASE>;<CAP-MIN>;IGNORE
+<T-S>		<ts>;<BASE>;<CAP-CAP>;IGNORE
+<T-s-h>		<tsh>;<BASE>;<CAP-MIN-MIN>;IGNORE
+<T-s-H>		<tsh>;<BASE>;<CAP-MIN-CAP>;IGNORE
+<T-S-h>		<tsh>;<BASE>;<CAP-CAP-MIN>;IGNORE
+<T-S-H>		<tsh>;<BASE>;<CAP-CAP-CAP>;IGNORE
+reorder-after <U0074>
+<t-h>		<th0>;<BASE>;<MIN-MIN>;IGNORE
+<t-H>		<th0>;<BASE>;<MIN-CAP>;IGNORE
+<t-s>		<ts>;<BASE>;<MIN-MIN>;IGNORE
+<t-S>		<ts>;<BASE>;<MIN-CAP>;IGNORE
+<t-s-h>		<tsh>;<BASE>;<MIN-MIN-MIN>;IGNORE
+<t-s-H>		<tsh>;<BASE>;<MIN-MIN-CAP>;IGNORE
+<t-S-h>		<tsh>;<BASE>;<MIN-CAP-MIN>;IGNORE
+<t-S-H>		<tsh>;<BASE>;<MIN-CAP-CAP>;IGNORE
+
+reorder-after <U0058>
+<X-h>		<xh>;<BASE>;<CAP-MIN>;IGNORE
+<X-H>		<xh>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0078>
+<x-h>		<xh>;<BASE>;<MIN-MIN>;IGNORE
+<x-H>		<xh>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-end
+
 END LC_COLLATE
 
 LC_MONETARY
--- libc/localedata/locales/tn_ZA~	2004-11-12 20:51:05.000000000 +0100
+++ libc/localedata/locales/tn_ZA	2004-11-12 20:51:05.000000000 +0100
@@ -52,8 +52,38 @@
 
 LC_COLLATE
 % Copy the template from ISO/IEC 14651 i.e.
-% use the rules there when making ordered lists of words.
 copy "iso14651_t1"
+
+% a b c d e e^ f g h i j k l m n o o^ p q r s s> t u v w x y z
+
+collating-symbol <e^>
+collating-symbol <o^>
+collating-symbol <s/>>
+
+reorder-after <BEFORE-F>
+<e^>
+reorder-after <BEFORE-P>
+<o^>
+reorder-after <BEFORE-R>
+<s/>>
+
+reorder-after <U0065>
+<U00EA> <e^>;<BASE>;<MIN>;IGNORE
+reorder-after <U0045>
+<U00CA> <e^>;<BASE>;<CAP>;IGNORE
+
+reorder-after <U006F>
+<U00F4> <o^>;<BASE>;<MIN>;IGNORE
+reorder-after <U004F>
+<U00D4> <o^>;<BASE>;<CAP>;IGNORE
+
+reorder-after <U0073>
+<U0161> <s/>>;<BASE>;<MIN>;IGNORE
+reorder-after <U0053>
+<U0160> <s/>>;<BASE>;<CAP>;IGNORE
+
+reorder-end
+
 END LC_COLLATE
 
 LC_MONETARY
--- libc/localedata/locales/ve_ZA~	2004-11-12 20:26:49.000000000 +0100
+++ libc/localedata/locales/ve_ZA	2004-11-12 20:44:23.000000000 +0100
@@ -54,8 +54,54 @@
 
 LC_COLLATE
 % Copy the template from ISO/IEC 14651 i.e.
-% use the rules there when making ordered lists of words.
 copy "iso14651_t1"
+
+% a b c d^ d e f g h i j k l^ l m n^ n n. o p q r s t^ t u v w x y z
+
+collating-symbol <d^>
+collating-symbol <l^>
+collating-symbol <n^>
+collating-symbol <n.>
+collating-symbol <t^>
+
+reorder-after <BEFORE-D>
+<d^>
+reorder-after <BEFORE-L>
+<l^>
+reorder-after <BEFORE-N>
+<n^>
+reorder-after <BEFORE-O>
+<n.>
+reorder-after <BEFORE-T>
+<t^>
+
+reorder-after <U0063>
+<U1E13> <d^>;<BASE>;<MIN>;IGNORE
+reorder-after <U0043>
+<U1E12> <d^>;<BASE>;<CAP>;IGNORE
+
+reorder-after <U006B>
+<U1E3D> <l^>;<BASE>;<MIN>;IGNORE
+reorder-after <U004B>
+<U1E3C> <l^>;<BASE>;<CAP>;IGNORE
+
+reorder-after <U006D>
+<U1E4B> <n^>;<BASE>;<MIN>;IGNORE
+reorder-after <U004D>
+<U1E4A> <n^>;<BASE>;<CAP>;IGNORE
+
+reorder-after <U006E>
+<U1E45> <n.>;<BASE>;<MIN>;IGNORE
+reorder-after <U004E>
+<U1E44> <n.>;<BASE>;<CAP>;IGNORE
+
+reorder-after <U0073>
+<U1E71> <t^>;<BASE>;<MIN>;IGNORE
+reorder-after <U0053>
+<U1E70> <t^>;<BASE>;<CAP>;IGNORE
+
+reorder-end
+
 END LC_COLLATE
 
 LC_MONETARY
--- libc/localedata/locales/xh_ZA.orig	Fri Nov 12 19:26:47 2004
+++ libc/localedata/locales/xh_ZA	Fri Nov 12 19:26:47 2004
@@ -52,9 +52,263 @@
 END LC_CTYPE
 
 LC_COLLATE
-% Copy the template from ISO/IEC 14651 i.e.
-% use the rules there when making ordered lists of words.
 copy "iso14651_t1"
+
+%% a b c ch d dl e f g gc gq h hl i j k kh l m n o p ph q qh
+%% r rh s t th ts tsh ty u v w x xh xq y z
+
+collating-symbol  <ch>
+collating-element <C-H> from "<U0043><U0048>"
+collating-element <C-h> from "<U0043><U0068>"
+collating-element <c-H> from "<U0063><U0048>"
+collating-element <c-h> from "<U0063><U0068>"
+
+collating-symbol  <dl>
+collating-element <D-L> from "<U0044><U004C>"
+collating-element <D-l> from "<U0044><U006C>"
+collating-element <d-L> from "<U0064><U004C>"
+collating-element <d-l> from "<U0064><U006C>"
+
+collating-symbol  <gc>
+collating-element <G-C> from "<U0047><U0043>"
+collating-element <G-c> from "<U0047><U0063>"
+collating-element <g-C> from "<U0067><U0043>"
+collating-element <g-c> from "<U0067><U0063>"
+
+collating-symbol  <gq>
+collating-element <G-Q> from "<U0047><U0051>"
+collating-element <G-q> from "<U0047><U0071>"
+collating-element <g-Q> from "<U0067><U0051>"
+collating-element <g-q> from "<U0067><U0071>"
+
+collating-symbol  <hl>
+collating-element <H-L> from "<U0048><U004C>"
+collating-element <H-l> from "<U0048><U006C>"
+collating-element <h-L> from "<U0068><U004C>"
+collating-element <h-l> from "<U0068><U006C>"
+
+collating-symbol  <kh>
+collating-element <K-H> from "<U004B><U0048>"
+collating-element <K-h> from "<U004B><U0068>"
+collating-element <k-H> from "<U006B><U0048>"
+collating-element <k-h> from "<U006B><U0068>"
+
+collating-symbol  <kl>
+collating-element <K-L> from "<U004B><U004C>"
+collating-element <K-l> from "<U004B><U006C>"
+collating-element <k-L> from "<U006B><U004C>"
+collating-element <k-l> from "<U006B><U006C>"
+
+collating-symbol  <ph>
+collating-element <P-H> from "<U0050><U0048>"
+collating-element <P-h> from "<U0050><U0068>"
+collating-element <p-H> from "<U0070><U0048>"
+collating-element <p-h> from "<U0070><U0068>"
+
+collating-symbol  <qh>
+collating-element <Q-H> from "<U0051><U0048>"
+collating-element <Q-h> from "<U0051><U0068>"
+collating-element <q-H> from "<U0071><U0048>"
+collating-element <q-h> from "<U0071><U0068>"
+
+collating-symbol  <rh>
+collating-element <R-H> from "<U0052><U0048>"
+collating-element <R-h> from "<U0052><U0068>"
+collating-element <r-H> from "<U0072><U0048>"
+collating-element <r-h> from "<U0072><U0068>"
+
+% symbol <th> is already used in iso14651_t1 
+collating-symbol  <th0>
+collating-element <T-H> from "<U0054><U0048>"
+collating-element <T-h> from "<U0054><U0068>"
+collating-element <t-H> from "<U0074><U0048>"
+collating-element <t-h> from "<U0074><U0068>"
+
+collating-symbol  <ts>
+collating-element <T-S> from "<U0054><U0053>"
+collating-element <T-s> from "<U0054><U0073>"
+collating-element <t-S> from "<U0074><U0053>"
+collating-element <t-s> from "<U0074><U0073>"
+
+collating-symbol  <tsh>
+collating-element <T-S-H> from "<U0054><U0053><U0048>"
+collating-element <T-S-h> from "<U0054><U0053><U0068>"
+collating-element <T-s-H> from "<U0054><U0073><U0048>"
+collating-element <T-s-h> from "<U0054><U0073><U0068>"
+collating-element <t-S-H> from "<U0074><U0053><U0048>"
+collating-element <t-S-h> from "<U0074><U0053><U0068>"
+collating-element <t-s-H> from "<U0074><U0073><U0048>"
+collating-element <t-s-h> from "<U0074><U0073><U0068>"
+
+collating-symbol  <ty>
+collating-element <T-Y> from "<U0054><U0059>"
+collating-element <T-y> from "<U0054><U0079>"
+collating-element <t-Y> from "<U0074><U0059>"
+collating-element <t-y> from "<U0074><U0079>"
+
+collating-symbol  <xh>
+collating-element <X-H> from "<U0058><U0048>"
+collating-element <X-h> from "<U0058><U0068>"
+collating-element <x-H> from "<U0078><U0048>"
+collating-element <x-h> from "<U0078><U0068>"
+
+collating-symbol  <xq>
+collating-element <X-Q> from "<U0058><U0051>"
+collating-element <X-q> from "<U0058><U0071>"
+collating-element <x-Q> from "<U0078><U0051>"
+collating-element <x-q> from "<U0078><U0071>"
+
+collating-symbol <CAP-CAP>
+collating-symbol <CAP-MIN>
+collating-symbol <MIN-CAP>
+collating-symbol <MIN-MIN>
+collating-symbol <CAP-CAP-CAP>
+collating-symbol <CAP-CAP-MIN>
+collating-symbol <CAP-MIN-CAP>
+collating-symbol <CAP-MIN-MIN>
+collating-symbol <MIN-CAP-CAP>
+collating-symbol <MIN-CAP-MIN>
+collating-symbol <MIN-MIN-CAP>
+collating-symbol <MIN-MIN-MIN>
+
+reorder-after <MIN>
+<MIN-MIN>
+<MIN-CAP>
+<MIN-MIN-MIN>
+<MIN-MIN-CAP>
+<MIN-CAP-MIN>
+<MIN-CAP-CAP>
+
+reorder-after <CAP>
+<CAP-MIN>
+<CAP-CAP>
+<CAP-MIN-MIN>
+<CAP-MIN-CAP>
+<CAP-CAP-MIN>
+<CAP-CAP-CAP>
+
+reorder-after <BEFORE-D>
+<ch>
+reorder-after <BEFORE-D>
+<dl>
+reorder-after <BEFORE-H>
+<gc>
+<gq>
+reorder-after <BEFORE-I>
+<hl>
+reorder-after <BEFORE-L>
+<kh>
+reorder-after <BEFORE-Q>
+<ph>
+reorder-after <BEFORE-R>
+<qh>
+reorder-after <BEFORE-S>
+<rh>
+reorder-after <BEFORE-U>
+<th0>
+<ts>
+<tsh>
+<ty>
+reorder-after <BEFORE-Y>
+<xh>
+<xq>
+
+reorder-after <U0043>
+<C-h>		<ch>;<BASE>;<CAP-MIN>;IGNORE
+<C-H>		<ch>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0063>
+<c-h>		<ch>;<BASE>;<MIN-MIN>;IGNORE
+<c-H>		<ch>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0044>
+<D-l>		<dl>;<BASE>;<CAP-MIN>;IGNORE
+<D-L>		<dl>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0064>
+<d-l>		<dl>;<BASE>;<MIN-MIN>;IGNORE
+<d-L>		<dl>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0047>
+<G-c>		<gc>;<BASE>;<CAP-MIN>;IGNORE
+<G-C>		<gc>;<BASE>;<CAP-CAP>;IGNORE
+<G-q>		<gq>;<BASE>;<CAP-MIN>;IGNORE
+<G-Q>		<gq>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0067>
+<g-c>		<gc>;<BASE>;<MIN-MIN>;IGNORE
+<g-C>		<gc>;<BASE>;<MIN-CAP>;IGNORE
+<g-q>		<gq>;<BASE>;<MIN-MIN>;IGNORE
+<g-Q>		<gq>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0048>
+<H-l>		<hl>;<BASE>;<CAP-MIN>;IGNORE
+<H-L>		<hl>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0068>
+<h-l>		<hl>;<BASE>;<MIN-MIN>;IGNORE
+<h-L>		<hl>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U004B>
+<K-h>		<kh>;<BASE>;<CAP-MIN>;IGNORE
+<K-H>		<kh>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U006B>
+<k-h>		<kh>;<BASE>;<MIN-MIN>;IGNORE
+<k-H>		<kh>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0050>
+<P-h>		<ph>;<BASE>;<CAP-MIN>;IGNORE
+<P-H>		<ph>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0070>
+<p-h>		<ph>;<BASE>;<MIN-MIN>;IGNORE
+<p-H>		<ph>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0051>
+<Q-h>		<qh>;<BASE>;<CAP-MIN>;IGNORE
+<Q-H>		<qh>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0071>
+<q-h>		<qh>;<BASE>;<MIN-MIN>;IGNORE
+<q-H>		<qh>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0052>
+<R-h>		<rh>;<BASE>;<CAP-MIN>;IGNORE
+<R-H>		<rh>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0072>
+<r-h>		<rh>;<BASE>;<MIN-MIN>;IGNORE
+<r-H>		<rh>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0054>
+<T-h>		<th0>;<BASE>;<CAP-MIN>;IGNORE
+<T-H>		<th0>;<BASE>;<CAP-CAP>;IGNORE
+<T-s>		<ts>;<BASE>;<CAP-MIN>;IGNORE
+<T-S>		<ts>;<BASE>;<CAP-CAP>;IGNORE
+<T-s-h>		<tsh>;<BASE>;<CAP-MIN-MIN>;IGNORE
+<T-s-H>		<tsh>;<BASE>;<CAP-MIN-CAP>;IGNORE
+<T-S-h>		<tsh>;<BASE>;<CAP-CAP-MIN>;IGNORE
+<T-S-H>		<tsh>;<BASE>;<CAP-CAP-CAP>;IGNORE
+<T-y>		<ty>;<BASE>;<CAP-MIN>;IGNORE
+<T-Y>		<ty>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0074>
+<t-h>		<th0>;<BASE>;<MIN-MIN>;IGNORE
+<t-H>		<th0>;<BASE>;<MIN-CAP>;IGNORE
+<t-s>		<ts>;<BASE>;<MIN-MIN>;IGNORE
+<t-S>		<ts>;<BASE>;<MIN-CAP>;IGNORE
+<t-s-h>		<tsh>;<BASE>;<MIN-MIN-MIN>;IGNORE
+<t-s-H>		<tsh>;<BASE>;<MIN-MIN-CAP>;IGNORE
+<t-S-h>		<tsh>;<BASE>;<MIN-CAP-MIN>;IGNORE
+<t-S-H>		<tsh>;<BASE>;<MIN-CAP-CAP>;IGNORE
+<t-y>		<ty>;<BASE>;<MIN-MIN>;IGNORE
+<t-Y>		<ty>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-after <U0058>
+<X-h>		<xh>;<BASE>;<CAP-MIN>;IGNORE
+<X-H>		<xh>;<BASE>;<CAP-CAP>;IGNORE
+<X-q>		<xq>;<BASE>;<CAP-MIN>;IGNORE
+<X-Q>		<xq>;<BASE>;<CAP-CAP>;IGNORE
+reorder-after <U0078>
+<x-h>		<xh>;<BASE>;<MIN-MIN>;IGNORE
+<x-H>		<xh>;<BASE>;<MIN-CAP>;IGNORE
+<x-q>		<xq>;<BASE>;<MIN-MIN>;IGNORE
+<x-Q>		<xq>;<BASE>;<MIN-CAP>;IGNORE
+
+reorder-end
+
 END LC_COLLATE
 
 LC_MONETARY
