--- glibc/sysdeps/unix/sysv/linux/dl-execstack.c.gotom	2005-08-18 20:55:21.000000000 +0900
+++ glibc/sysdeps/unix/sysv/linux/dl-execstack.c	2005-08-19 15:50:59.051597872 +0900
@@ -84,7 +84,8 @@
 	page -= size;
       else
 	{
-	  if (errno != ENOMEM)	/* Unexpected failure mode.  */
+	  /* Unexpected failure mode.  */
+	  if (errno != ENOMEM && errno != EFAULT)
 	    return errno;
 
 	  if (size == GLRO(dl_pagesize))
@@ -107,7 +108,8 @@
 	page += size;
       else
 	{
-	  if (errno != ENOMEM)	/* Unexpected failure mode.  */
+	  /* Unexpected failure mode.  */
+	  if (errno != ENOMEM && errno != EFAULT)
 	    return errno;
 
 	  if (size == GLRO(dl_pagesize))
