diff -uNr libc.orig/localedata/locales/af_ZA libc/localedata/locales/af_ZA
--- libc.orig/localedata/locales/af_ZA	2004-04-29 00:55:00.000000000 +0200
+++ libc/localedata/locales/af_ZA	2004-10-29 13:05:39.000000000 +0200
@@ -1,33 +1,45 @@
 escape_char  /
 comment_char  %
 
-% Afrikaans language locale for South Africa
-% Source: RAP
-% Email: schalkc@ntaba.co.za
-% Tel: +27 12 3293397
-% Fax: +27 12 3293397
-% Language: af
-% Territory: ZA
-% Revision: 0.2
-% Date: 1999-09-28
-% Users: general
-% Repertoiremap: mnemonic.ds
-% Charset: ISO-8859-1
-% Distribution and use is free, also
-% for commercial purposes.
+% ChangeLog
+% 1.2 (2004-10-28):
+%      2004-10-28 Dwayne Bailey <dwayne@translate.org.za>
+%        - Corrected country_name should be "Suid-Afrika"
+%      2004-10-27 Dwayne Bailey <dwayne@translate.org.za>
+%        - Change decimal and thousand separator to match common convention of
+%          [comma]==thousand, [point]==decimal.  Thus copy en_ZA
+%        - Add country_post, lang_lib
+%        - abmon March: Mar -> Mrt
+%        - name_* corrected all capitalisation
+%        - Add Translate.org.za to clarify "source"
+%        - Update all contact details
+%        - Change yesexpr to also accept yY
+%        - Update yes/noexpr to remove .*
+%      2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%        - Added name_miss
+% 1.1 (2003-04-23):
+%      2003-04-23 Dwayne Bailey <dwayne@translate.org.za>
+%        - Added LC_MEASUREMENT to LC_IDENTIFICATION
+%        - PAPER, TELEPHONE and MEASUREMENT copy en_ZA
+%        - Added "^" to yesexpr and noexpr ie "^[nN].*"
+%        - Change maintainer to Zuza Software Foundation
+%        - Add all missing items in LC_ADDRESS
+%        - Add all name_* to LC_NAME
+%        - Add various comments explaining items
+
 
 LC_IDENTIFICATION
 title      "Afrikaans locale for South Africa"
-source     "RAP"
-address    ""
-contact    ""
-email      "bug-glibc@gnu.org"
-tel        ""
-fax        ""
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "Box 28364, Sunnyside, 0132, South Africa"
+contact    "Dwayne Bailey"
+email      "dwayne@translate.org.za"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
 language   "Afrikaans"
 territory  "South Africa"
-revision   "1.0"
-date       "2000-06-29"
+revision   "1.2"
+date       "2004-10-28"
 %
 category  "af_ZA:2000";LC_IDENTIFICATION
 category  "af_ZA:2000";LC_CTYPE
@@ -35,11 +47,12 @@
 category  "af_ZA:2000";LC_TIME
 category  "af_ZA:2000";LC_NUMERIC
 category  "af_ZA:2000";LC_MONETARY
-category  "af_ZA:2000";LC_MESSAGES
-category  "af_ZA:2000";LC_PAPER
+category  "af_ZA:2003";LC_MESSAGES
+category  "af_ZA:2003";LC_PAPER
 category  "af_ZA:2000";LC_NAME
-category  "af_ZA:2000";LC_ADDRESS
-category  "af_ZA:2000";LC_TELEPHONE
+category  "af_ZA:2003";LC_ADDRESS
+category  "af_ZA:2003";LC_TELEPHONE
+category  "af_ZA:2003";LC_MEASUREMENT
 
 END LC_IDENTIFICATION
 
@@ -51,36 +64,24 @@
 copy "en_DK"
 END LC_COLLATE
 
-
 LC_MONETARY
-int_curr_symbol      "<U005A><U0041><U0052><U0020>"
-currency_symbol      "<U0052>"
-mon_decimal_point    "<U002C>"
-mon_thousands_sep    "<U002E>"
-mon_grouping         3;3
-positive_sign        ""
-negative_sign        "<U002D>"
-int_frac_digits      2
-frac_digits          2
-p_cs_precedes        1
-p_sep_by_space       0
-n_cs_precedes        1
-n_sep_by_space       0
-p_sign_posn          1
-n_sign_posn          1
+copy "en_ZA"
 END LC_MONETARY
 
 LC_NUMERIC
-decimal_point        "<U002C>"
-thousands_sep        "<U002E>"
-grouping             3;3
+copy "en_ZA"
 END LC_NUMERIC
 
 LC_TIME
+% abday - The abbreviations for the week days:
+% - So, Ma, Di, Wo, Do, Vr, Sa
 abday       "<U0053><U006F>";"<U004D><U0061>";/
             "<U0044><U0069>";"<U0057><U006F>";/
             "<U0044><U006F>";"<U0056><U0072>";/
             "<U0053><U0061>"
+
+% day - The full names of the week days:
+% Sondag, Maandag, Dinsdag, Woensdag, Donderdag, Vrydag, Saterdag
 day         "<U0053><U006F><U006E><U0064><U0061><U0067>";/
             "<U004D><U0061><U0061><U006E><U0064><U0061><U0067>";/
             "<U0044><U0069><U006E><U0073><U0064><U0061><U0067>";/
@@ -88,12 +89,19 @@
             "<U0044><U006F><U006E><U0064><U0065><U0072><U0064><U0061><U0067>";/
             "<U0056><U0072><U0079><U0064><U0061><U0067>";/
             "<U0053><U0061><U0074><U0065><U0072><U0064><U0061><U0067>"
+
+% abmon - The abbreviations for the months 
+% - Jan, Feb, Mrt, Apr, Mei, Jun, Jul, Aug, Sep, Okt, Nov, Des
 abmon       "<U004A><U0061><U006E>";"<U0046><U0065><U0062>";/
-            "<U004D><U0061><U0072>";"<U0041><U0070><U0072>";/
+            "<U004D><U0072><U0074>";"<U0041><U0070><U0072>";/
             "<U004D><U0065><U0069>";"<U004A><U0075><U006E>";/
             "<U004A><U0075><U006C>";"<U0041><U0075><U0067>";/
             "<U0053><U0065><U0070>";"<U004F><U006B><U0074>";/
             "<U004E><U006F><U0076>";"<U0044><U0065><U0073>"
+
+% mon - The full names of the months - 
+% Januarie, Februarie, Maart, April, Mei, Junie
+% Julie, Augustus, September, Oktober, November, Desember
 mon         "<U004A><U0061><U006E><U0075><U0061><U0072><U0069><U0065>";/
             "<U0046><U0065><U0062><U0072><U0075><U0061><U0072><U0069><U0065>";/
             "<U004D><U0061><U0061><U0072><U0074>";/
@@ -106,50 +114,170 @@
             "<U004F><U006B><U0074><U006F><U0062><U0065><U0072>";/
             "<U004E><U006F><U0076><U0065><U006D><U0062><U0065><U0072>";/
             "<U0044><U0065><U0073><U0065><U006D><U0062><U0065><U0072>"
+
+% Abbreviated date and time representation to be referenced by the "%c" field descriptor -
+%
+% "%a" (short weekday name),
+% "%d" (day of month as a decimal number),
+% "%b" (short month name),
+% "%Y" (year with century as a decimal number),
+% "%T" (24-hour clock time in format HH:MM:SS),
+% "%Z" (Time zone name)
 d_t_fmt     "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
-d_fmt       "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0079>"
+
+% Date representation to be referenced by the "%x" field descriptor -
+% "%d/%m/%Y", day/month/year as decimal numbers (01/01/2000).
+d_fmt       "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+
+% Time representation to be referenced by the "%X" field descriptor -
+% "%T" (24-hour clock time in format HH:MM:SS)
 t_fmt       "<U0025><U0054>"
+
+% Define representation of ante meridian and post maritime strings -
+% The "" mean 'default to "AM" and "PM".
 am_pm       "<U0056><U004D>";"<U004E><U004D>"
+
+% Define time representation in 12-hour format with "am_pm", to be referenced by the "%r"
+% The "" means that this format is not supported.
 t_fmt_ampm  ""
+
+% Date representation not described in ISO/IEC 14652. Comes out as -
+% "%a %b %e %H:%M:%S %Z %Y" which is default "date" command output
+%
+% %a - abbreviated weekday name,
+% %b - abbreviated month name,
+% %e - day of month as a decimal number with leading space (1 to 31),
+% %H - hour (24-hour clock) as a decimal number (00 to 23),
+% %M - minute as a decimal number (00 to 59),
+% %S - seconds as a decimal number (00 to 59),
+% %Z - time-zone name,
+% %Y - year with century as a decimal number,e.g. 2001.
 date_fmt	"<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
 <U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
 <U0025><U005A><U0020><U0025><U0059>"
 END LC_TIME
 
 LC_MESSAGES
-yesexpr     "<U005E><U005B><U006A><U004A><U005D><U002E><U002A>"
-noexpr      "<U005E><U005B><U006E><U004E><U005D><U002E><U002A>"
+% The affirmative response -
+% "^[jJyY]"
+yesexpr     "<U005E><U005B><U006A><U004A><U0079><U0059><U005D>"
+
+% The negative response -
+% "^[nN]"
+noexpr      "<U005E><U005B><U006E><U004E><U005D>"
 END LC_MESSAGES
 
 LC_PAPER
-% FIXME
-height   297
-% FIXME
-width    210
+copy "en_ZA"
 END LC_PAPER
 
 LC_TELEPHONE
-tel_int_fmt    "<U002B><U0025><U0063><U0020><U0025><U0061><U0020><U0025>/
-<U006C>"
-int_prefix     "<U0032><U0037>"
+copy "en_ZA"
 END LC_TELEPHONE
 
 LC_MEASUREMENT
-% FIXME
-measurement    1
+copy "en_ZA"
 END LC_MEASUREMENT
 
 LC_NAME
+% Format for addressing a person.
+% "%d%t%g%t%m%t%f"
+%
+% "Salutation",
+% "Empty string, or <Space>",
+% "First given name",
+% "Empty string, or <Space>",
+% "Middle names",
+% "Empty string, or <Space>",
+% "Clan names"
 name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
 <U0025><U006D><U0025><U0074><U0025><U0066>"
+
+% General salutation for any sex
+% (no term)
+name_gen    ""
+
+% Salutation for unmarried females
+% "mej"
+name_miss   "<U006D><U0065><U006A>"
+ 
+% Salutation for males
+% "mnr"
+name_mr     "<U006D><U006E><U0072>"
+
+% Salutation for married females
+% "mev"
+name_mrs    "<U006D><U0065><U0076>"
+
+% Salutation valid for all females
+% "me"
+name_ms     "<U006D><U0065>"
 END LC_NAME
 
 LC_ADDRESS
+% Representation of postal addresses (minus the addressee's name) in South
+% Africa. (Ignored for now)
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N", which gives -
+%
+% "firm name",
+% "end of line",
+% "C/O address",
+% "end of line",
+% "department name",
+% "Building name",
+% "end of line",
+% "street or block name",
+% "space",
+% "house number or designation",
+% "space",
+% "floor number",
+% "space",
+% "room number, door designation",
+% "end of line",
+% "postal code",
+% "space",
+% "town, city",
+% "end of line",
+% "country designation for the <country_post> keyword",
+% "end of line
 postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
 <U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
 <U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
 <U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
 <U004E><U0025><U0063><U0025><U004E>"
+
+% Country name in Afrikaans - "Suid-Afrika"
+country_name  "<U0053><U0075><U0069><U0064><U002D>/
+<U0041><U0066><U0072><U0069><U006B><U0061>"
+
+% Language name in Afrikaans - "Afrikaans"
+lang_name     "<U0041><U0066><U0072><U0069><U006B><U0061><U0061><U006E><U0073>"
+
+% CEPT MAILCODES are suggested
+% Alternatively use the code found on your countries postal item tracking number
+% "ZA"
+country_post   "<U005A><U0041>"
+
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% ISO 2108
+% http://www.isbn-international.org/html/prefix/prefa.htm
+country_isbn  0
+
+% ISO 639 language abbreviations:
+% 639-1 2 letter, 639-2 3 letter terminology
+% http://www.loc.gov/standards/iso639-2/englangn.html
+% "af", "afr", "afr"
+lang_ab       "<U0061><U0066>"
+lang_term     "<U0061><U0066><U0072>"
+lang_lib      "<U0061><U0066><U0072>"
+
+% ISO 3166 country number and 2 and 3 letter abbreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
 country_ab2   "<U005A><U0041>"
 country_ab3   "<U005A><U0041><U0046>"
 country_num   710
diff -uNr libc.orig/localedata/locales/en_ZA libc/localedata/locales/en_ZA
--- libc.orig/localedata/locales/en_ZA	2004-01-01 20:53:38.000000000 +0100
+++ libc/localedata/locales/en_ZA	2004-10-27 16:55:53.000000000 +0200
@@ -2,14 +2,14 @@
 comment_char  %
 
 % English language locale for South Africa
-% Source: Zuza Software Foundation
+% Source: Zuza Software Foundation (Translate.org.za)
 % Email: dwayne@translate.org.za
-% Tel: +27 21 4487827
-% Fax: +27 21 4489574
+% Tel: +27 12 3430389
+% Fax: +27 12 3430389
 % Language: en
 % Territory: ZA
-% Revision: 1.1
-% Date: 2003-03-14
+% Revision: 1.2
+% Date: 2004-10-27
 % Users: general
 % Repertoiremap: mnemonic,ds
 % Charset: ISO-8859-1
@@ -17,6 +17,20 @@
 % for commercial purposes.
 %
 % Changelog
+% 1.2 (2004-10-27):
+%     2004-10-27 Dwayne Bailey <dwayne@translate.org.za>
+%       - Remove .* from LC_MESSAGES yes/noexpr
+%       - Change contact details and add Translate.org.za
+%         to project name
+%     2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%       - Added country_post
+%       - Added abbreviation to LC_IDENTIFICATION
+%     2003-05-17 Dwayne Bailey <dwayne@translate.org.za>
+%       - Add lang_ab, lang_term
+%       - Add various comments for reference documents
+%       - Add country_isbn, country_car, country_ab2, 
+%         country_ab3, country_num
+%       - Add name_* for all salutations
 % 1.1 (2003-03-14):
 %   - Updated maintainer to Zuza Software Foundation
 %   - Changed %x for LC_TIME to use dd/mm/ccyy
@@ -27,16 +41,19 @@
 
 LC_IDENTIFICATION
 title      "English locale for South Africa"
-source     "Zuza Software Foundation"
-address    "Box 13412, Mowbray, 7701, South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "Box 28364, Sunnyside, 0132, South Africa"
 contact    "Dwayne Bailey"
 email      "dwayne@translate.org.za"
-tel        "+27 21 448 7827"
-fax        "+27 21 448 9574"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
 language   "English"
 territory  "South Africa"
-revision   "1.1"
-date       "2003-03-14"
+% audience     ""
+% application  ""
+abbreviation "Translate.org.za"
+revision   "1.2"
+date       "2004-10-27"
 %
 category  "en_ZA:2003";LC_IDENTIFICATION
 category  "en_ZA:2000";LC_CTYPE
@@ -47,7 +64,7 @@
 category  "en_ZA:2003";LC_MESSAGES
 category  "en_ZA:2000";LC_PAPER
 category  "en_ZA:2000";LC_MEASUREMENT
-category  "en_ZA:2000";LC_NAME
+category  "en_ZA:2003";LC_NAME
 category  "en_ZA:2003";LC_ADDRESS
 category  "en_ZA:2003";LC_TELEPHONE
 
@@ -62,12 +79,23 @@
 END LC_COLLATE
 
 LC_MONETARY
+% ISO 4217 Currency and fund codes
+% http://www.bsi-global.com/Technical+Information/Publications/_Publications/tig90.xalter
+% "ZAR "
 int_curr_symbol         "<U005A><U0041><U0052><U0020>"
+
+% "R"
 currency_symbol         "<U0052>"
+
+% "."
 mon_decimal_point       "<U002E>"
+
+% ","
 mon_thousands_sep       "<U002C>"
 mon_grouping            3;3
 positive_sign           ""
+
+% "-"
 negative_sign           "<U002D>"
 int_frac_digits         2
 frac_digits             2
@@ -80,16 +108,24 @@
 END LC_MONETARY
 
 LC_NUMERIC
+% "."
 decimal_point           "<U002E>"
+
+% ","
 thousands_sep           "<U002C>"
 grouping                3;3
 END LC_NUMERIC
 
 LC_TIME
+% abday - The abbreviations for the week days:
+% - Sun, Mon, Tue, Wed, Thu, Fri, Sat
 abday       "<U0053><U0075><U006E>";"<U004D><U006F><U006E>";/
             "<U0054><U0075><U0065>";"<U0057><U0065><U0064>";/
             "<U0054><U0068><U0075>";"<U0046><U0072><U0069>";/
             "<U0053><U0061><U0074>"
+
+% day - The full names of the week days:
+% - Sunday, Monday, Tuesday, Wednesday, Thursday, Friday, Saturday
 day         "<U0053><U0075><U006E><U0064><U0061><U0079>";/
             "<U004D><U006F><U006E><U0064><U0061><U0079>";/
             "<U0054><U0075><U0065><U0073><U0064><U0061><U0079>";/
@@ -97,12 +133,19 @@
             "<U0054><U0068><U0075><U0072><U0073><U0064><U0061><U0079>";/
             "<U0046><U0072><U0069><U0064><U0061><U0079>";/
             "<U0053><U0061><U0074><U0075><U0072><U0064><U0061><U0079>"
+
+% abmon - The abbreviations for the months 
+% - Jan, Feb, Mar, Apr, May, Jun, Jul, Aug, Sep, Oct, Nov, Dec
 abmon       "<U004A><U0061><U006E>";"<U0046><U0065><U0062>";/
             "<U004D><U0061><U0072>";"<U0041><U0070><U0072>";/
             "<U004D><U0061><U0079>";"<U004A><U0075><U006E>";/
             "<U004A><U0075><U006C>";"<U0041><U0075><U0067>";/
             "<U0053><U0065><U0070>";"<U004F><U0063><U0074>";/
             "<U004E><U006F><U0076>";"<U0044><U0065><U0063>"
+
+% mon - The full names of the months - 
+% - January, February, March, April, May, June
+%   July, August, September, October, November, December
 mon         "<U004A><U0061><U006E><U0075><U0061><U0072><U0079>";/
             "<U0046><U0065><U0062><U0072><U0075><U0061><U0072><U0079>";/
             "<U004D><U0061><U0072><U0063><U0068>";/
@@ -115,19 +158,57 @@
             "<U004F><U0063><U0074><U006F><U0062><U0065><U0072>";/
             "<U004E><U006F><U0076><U0065><U006D><U0062><U0065><U0072>";/
             "<U0044><U0065><U0063><U0065><U006D><U0062><U0065><U0072>"
+
+% Abreviated date and time representation to be referenced by the "%c" field descriptor -
+%
+% "%a" (short weekday name),
+% "%d" (day of month as a decimal number),
+% "%b" (short month name),
+% "%Y" (year with century as a decimal number),
+% "%T" (24-hour clock time in format HH:MM:SS),
+% "%Z" (Time zone name)
 d_t_fmt     "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+
+% Date representation to be referenced by the "%x" field descriptor -
+% "%d/%m/%Y", day/month/year as decimal numbers (01/01/2000).
 d_fmt       "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+
+% Time representation to be referenced by the "%X" field descriptor -
+% "%T" (24-hour clock time in format HH:MM:SS)
 t_fmt       "<U0025><U0054>"
+
+% Define representation of ante meridiem and post meridiem strings -
+% The "" mean default to "AM" and "PM".
 am_pm       "";""
+
+% Define time representation in 12-hour format with "am_pm", to be referenced by the "%r"
+% The "" means that this format is not supported.
 t_fmt_ampm  ""
+
+% Date representation not described in ISO/IEC 14652. Comes out as -
+% "%a %b %e %H:%M:%S %Z %Y" which is default "date" command output
+%
+% %a - abbreviated weekday name,
+% %b - abreviated month name,
+% %e - day of month as a decimal number with leading space (1 to 31),
+% %H - hour (24-hour clock) as a decimal number (00 to 23),
+% %M - minute as a decimal number (00 to 59),
+% %S - seconds as a decimal number (00 to 59),
+% %Z - time-zone name,
+% %Y - year with century as a decimal number,e.g. 2001.
 date_fmt	"<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
 <U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
 <U0025><U005A><U0020><U0025><U0059>"
 END LC_TIME
 
 LC_MESSAGES
-yesexpr     "<U005E><U005B><U0079><U0059><U005D><U002E><U002A>"
-noexpr      "<U005E><U005B><U006E><U004E><U005D><U002E><U002A>"
+% The affirmative response -
+% "^[yY]"
+yesexpr     "<U005E><U005B><U0079><U0059><U005D>"
+
+% The negative response -
+% "^[nN]"
+noexpr      "<U005E><U005B><U006E><U004E><U005D>"
 END LC_MESSAGES
 
 LC_PAPER
@@ -136,10 +217,26 @@
 END LC_PAPER
 
 LC_TELEPHONE
+% Representation of telephone number for international use -
+% "+%c %a %l", which is
+% "+%c - country code",
+% "%a - area code without the prefix (often 0)",
+% "%l local number".
 tel_int_fmt    "<U002B><U0025><U0063><U0020><U0025><U0061><U0020><U0025>/
 <U006C>"
+
+% Represntation of telephone number for domestic use -
+% "(%A) %l", which is
+% %A - area code with national prefix
+% %l - local number".
 tel_dom_fmt    "<U0028><U0025><U0041><U0029><U0020><U0025><U006C>"
+
+% Prefix for making international calls
+% "09"
 int_select     "<U0030><U0039>"
+
+% International dialing code
+% "27"
 int_prefix     "<U0032><U0037>"
 
 END LC_TELEPHONE
@@ -149,20 +246,106 @@
 END LC_MEASUREMENT
 
 LC_NAME
+% Format for addressing a person.
+% "%d%t%g%t%m%t%f"
+%
+% "Salutation",
+% "Empty string, or <Space>",
+% "First given name",
+% "Empty string, or <Space>",
+% "Middle names",
+% "Empty string, or <Space>",
+% "Clan names"
 name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
 <U0025><U006D><U0025><U0074><U0025><U0066>"
+
+% General salutation for any sex
+% (no term)
+name_gen    ""
+
+% Salutation for unmarried females
+% "Miss"
+name_miss   "<U004D><U0069><U0073><U0073>"
+ 
+% Salutation for males
+% "Mr"
+name_mr     "<U004D><U0072>"
+
+% Salutation for married females
+% "Mrs"
+name_mrs    "<U004D><U0072><U0073>"
+
+% Salutation valid for all females
+% "Ms"
+name_ms     "<U004D><U0073>"
 END LC_NAME
 
 LC_ADDRESS
+% Representation of postal addresses (minus the addressee's name) in South
+% Africa. (Ignored for now)
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N", which gives -
+%
+% "firm name",
+% "end of line",
+% "C/O address",
+% "end of line",
+% "department name",
+% "Building name",
+% "end of line",
+% "street or block name",
+% "space",
+% "house number or designation",
+% "space",
+% "floor number",
+% "space",
+% "room number, door designation",
+% "end of line",
+% "postal code",
+% "space",
+% "town, city",
+% "end of line",
+% "country designation for the <country_post> keyword",
+% "end of line
 postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
 <U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
 <U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
 <U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
 <U004E><U0025><U0063><U0025><U004E>"
+
+% Country name in English - "South Africa"
 country_name  "<U0053><U006F><U0075><U0074><U0068><U0020>/
 <U0041><U0066><U0072><U0069><U0063><U0061>"
+
+% Language name in English - "English"
 lang_name     "<U0045><U006E><U0067><U006C><U0069><U0073><U0068>"
+
+% CEPT MAILCODES are suggested
+% Alternatively use the code found on your countries postal item tracking number
+% "ZA"
+country_post   "<U005A><U0041>"
+
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% ISO 2108
+% http://www.isbn-international.org/html/prefix/prefa.htm
+country_isbn  0
+
+% ISO 639 language abbreviations:
+% 639-1 2 letter, 639-2 3 letter terminology
+% http://www.loc.gov/standards/iso639-2/englagn.html
+% "en", "eng"
+lang_ab       "<U0065><U006E>"
+lang_term     "<U0065><U006E><U0067>"
+lang_lib      "<U0065><U006E><U0067>"
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
 country_ab2   "<U005A><U0041>"
 country_ab3   "<U005A><U0041><U0046>"
 country_num   710
+
 END LC_ADDRESS
diff -uNr libc.orig/localedata/locales/nr_ZA libc/localedata/locales/nr_ZA
--- libc.orig/localedata/locales/nr_ZA	1970-01-01 01:00:00.000000000 +0100
+++ libc/localedata/locales/nr_ZA	2004-11-09 22:04:05.000000000 +0100
@@ -0,0 +1,273 @@
+escape_char  /
+comment_char  %
+
+% ChangeLog
+% 0.2 (2004-11-09):
+%     2004-11-09 Dwayne Bailey <dwayne@translate.org.za>
+%       - <UNNNN> escape everything
+%       - Update all contact information
+%       - Change %d to %-e for d_t_fmt and date_fmt
+%       - Reset yes/noexpr to English
+%       - set all abmon, mon, abday, day to English until correct defintions
+%         are available.
+%     2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%       - Added country_ab2/3, country_num
+%     2004-02-27 Dwayne Bailey <dwayne@translate.org.za>
+%	    - Correct capatilisation of lang_name
+% 0.1 (2004-02-24):
+%       - Initial Southern Ndebele locale for South Africa
+%         by Zuza Software Foundation
+
+LC_IDENTIFICATION
+title      "Southern Ndebele locale for South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "PO Box 28364, Sunnyside, 0132, South Africa"
+contact    "Dwayne Bailey"
+email      "dwayne@translate.org.za"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
+language   "Southern Ndebele"
+territory  "South Africa"
+revision   "0.2"
+date       "2004-11-09"
+%
+category  "nr_ZA:2004";LC_IDENTIFICATION
+category  "nr_ZA:2004";LC_CTYPE
+category  "nr_ZA:2004";LC_COLLATE
+category  "nr_ZA:2004";LC_TIME
+category  "nr_ZA:2004";LC_NUMERIC
+category  "nr_ZA:2004";LC_MONETARY
+category  "nr_ZA:2004";LC_MESSAGES
+category  "nr_ZA:2004";LC_PAPER
+category  "nr_ZA:2004";LC_NAME
+category  "nr_ZA:2004";LC_ADDRESS
+category  "nr_ZA:2004";LC_TELEPHONE
+category  "nr_ZA:2004";LC_MEASUREMENT
+
+END LC_IDENTIFICATION
+
+LC_CTYPE
+% Use the characters described in the charmap file "i18n.tgz"
+copy "i18n"
+END LC_CTYPE
+
+LC_COLLATE
+% Copy the template from ISO/IEC 14651 i.e.
+% use the rules there when making ordered lists of words.
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_MONETARY
+copy "en_ZA"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "en_ZA"
+END LC_NUMERIC
+
+LC_TIME
+
+% abday - The abbreviations for the week days:
+% - Sun, Mon, Tue, Wed, Thu, Fri, Sat
+abday       "<U0053><U0075><U006E>";"<U004D><U006F><U006E>";/
+            "<U0054><U0075><U0065>";"<U0057><U0065><U0064>";/
+            "<U0054><U0068><U0075>";"<U0046><U0072><U0069>";/
+            "<U0053><U0061><U0074>"
+ 
+% day - The full names of the week days:
+% - Sunday, Monday, Tuesday, Wednesday, Thursday, Friday, Saturday
+day         "<U0053><U0075><U006E><U0064><U0061><U0079>";/
+            "<U004D><U006F><U006E><U0064><U0061><U0079>";/
+            "<U0054><U0075><U0065><U0073><U0064><U0061><U0079>";/
+            "<U0057><U0065><U0064><U006E><U0065><U0073><U0064><U0061><U0079>";/
+            "<U0054><U0068><U0075><U0072><U0073><U0064><U0061><U0079>";/
+            "<U0046><U0072><U0069><U0064><U0061><U0079>";/
+            "<U0053><U0061><U0074><U0075><U0072><U0064><U0061><U0079>"
+ 
+% abmon - The abbreviations for the months 
+% - Jan, Feb, Mar, Apr, May, Jun, Jul, Aug, Sep, Oct, Nov, Dec
+abmon       "<U004A><U0061><U006E>";"<U0046><U0065><U0062>";/
+            "<U004D><U0061><U0072>";"<U0041><U0070><U0072>";/
+            "<U004D><U0061><U0079>";"<U004A><U0075><U006E>";/
+            "<U004A><U0075><U006C>";"<U0041><U0075><U0067>";/
+            "<U0053><U0065><U0070>";"<U004F><U0063><U0074>";/
+            "<U004E><U006F><U0076>";"<U0044><U0065><U0063>"
+ 
+% mon - The full names of the months - 
+% - January, February, March, April, May, June
+%   July, August, September, October, November, December
+mon         "<U004A><U0061><U006E><U0075><U0061><U0072><U0079>";/
+            "<U0046><U0065><U0062><U0072><U0075><U0061><U0072><U0079>";/
+            "<U004D><U0061><U0072><U0063><U0068>";/
+            "<U0041><U0070><U0072><U0069><U006C>";/
+            "<U004D><U0061><U0079>";/
+            "<U004A><U0075><U006E><U0065>";/
+            "<U004A><U0075><U006C><U0079>";/
+            "<U0041><U0075><U0067><U0075><U0073><U0074>";/
+            "<U0053><U0065><U0070><U0074><U0065><U006D><U0062><U0065><U0072>";/
+            "<U004F><U0063><U0074><U006F><U0062><U0065><U0072>";/
+            "<U004E><U006F><U0076><U0065><U006D><U0062><U0065><U0072>";/
+            "<U0044><U0065><U0063><U0065><U006D><U0062><U0065><U0072>"
+
+% Abreviated date and time representation to be referenced by the "%c" field descriptor -
+d_t_fmt     "<U0025><U0061><U0020><U0025><U002D><U0065><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+%
+% "%a" (short weekday name),
+% "%-e" (day of month as a decimal number),
+% "%b" (short month name),
+% "%Y" (year with century as a decimal number),
+% "%T" (24-hour clock time in format HH:MM:SS),
+% "%Z" (Time zone name)
+ 
+% Date representation to be referenced by the "%x" field descriptor -
+d_fmt   "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+% "%d/%m/%Y", day/month/year as decimal numbers (01/01/2000).
+ 
+% Time representation to be referenced by the "%X" field descriptor -
+t_fmt       "<U0025><U0054>"
+% "%T" (24-hour clock time in format HH:MM:SS)
+ 
+% Define representation of ante meridiem and post meridiem strings -
+am_pm       "";""
+% The "" mean 'default to "AM" and "PM".
+ 
+% Define time representation in 12-hour format with "am_pm", to be referenced by the "%r"
+t_fmt_ampm  ""
+% The "" means that this format is not supported.
+ 
+% Date representation not described in ISO/IEC 14652. Comes out as -
+% "%a %b %-e %H:%M:%S %Z %Y" which is default "date" command output
+date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U002D><U0065>/
+<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U005A><U0020><U0025><U0059>"
+%
+% %a - abbreviated weekday name,
+% %b - abreviated month name,
+% %-e - day of month as a decimal number with leading space (1 to 31),
+% %H - hour (24-hour clock) as a decimal number (00 to 23),
+% %M - minute as a decimal number (00 to 59),
+% %S - seconds as a decimal number (00 to 59),
+% %Z - time-zone name,
+% %Y - year with century as a decimal number,e.g. 2001.
+ 
+END LC_TIME
+
+LC_MESSAGES
+% FIXME: Check both of these
+% The affirmative response - TODO
+% "^[yY]"
+yesexpr     "<U005E><U005B><U0079><U0059><U005D>"
+
+% The negative response - TODO
+% "^[nN]"
+noexpr      "<U005E><U005B><U006E><U004E><U005D>"
+END LC_MESSAGES
+
+LC_PAPER
+copy "en_ZA"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "en_ZA"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "en_ZA"
+END LC_MEASUREMENT
+
+LC_NAME
+% Format for addressing a person.
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+% "%d%t%g%t%m%t%f"
+% "Salutation",
+% "Empty string, or <Space>",
+% "First given name",
+% "Empty string, or <Space>",
+% "Middle names",
+% "Empty string, or <Space>",
+% "Clan names"
+ 
+% FIXME - define all the following name_*
+% General salutation for any sex
+% name_gen    ""
+
+% Salutation for unmarried females - ""
+% name_miss   ""
+ 
+% Salutation for males - ""
+% name_mr     ""
+
+% Salutation for married females - ""
+% name_mrs    ""
+
+% Salutation valid for all females - "" (no term)
+% name_ms     ""
+
+END LC_NAME
+
+LC_ADDRESS
+% Country name in Southern Ndebele - "iSewula Afrika"
+country_name "<U0069><U0053><U0065><U0077><U0075><U006C><U0061><U0020><U0041><U0066><U0072><U0069><U006B><U0061>"
+ 
+% Abbreviated country postal name - "ZA"
+country_post "<U005A><U0041>"
+ 
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% FIXME define the following correctly
+% country_isbn ""
+% country_num 
+ 
+% Language name in Southern Ndebele - "IsiNdebele"
+lang_name "<U0049><U0073><U0069><U004E><U0064><U0065><U0062><U0065><U006C><U0065>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+% "nr", "nbl"
+lang_ab   "<U006E><U0072>"
+lang_term "<U006E><U0062><U006C>"
+ 
+% Representation of postal addresses (minus the addressee's name) in South
+% Africa. (Ignored for now)
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N", which gives -
+% "firm name",
+% "end of line",
+% "C/O address",
+% "end of line",
+% "department name",
+% "Building name",
+% "end of line",
+% "street or block name",
+% "space",
+% "house number or designation",
+% "space",
+% "floor number",
+% "space",
+% "room number, door designation",
+% "end of line",
+% "postal code",
+% "space",
+% "town, city",
+% "end of line",
+% "country designation for the <country_post> keyword",
+% "end of line
+%
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
+END LC_ADDRESS
+
diff -uNr libc.orig/localedata/locales/nso_ZA libc/localedata/locales/nso_ZA
--- libc.orig/localedata/locales/nso_ZA	1970-01-01 01:00:00.000000000 +0100
+++ libc/localedata/locales/nso_ZA	2004-10-27 22:48:11.000000000 +0200
@@ -0,0 +1,273 @@
+escape_char  /
+comment_char  %
+
+% ChangeLog
+% 0.2 (2004-10-27):
+%      2004-10-27 Dwayne Bailey <dwayne@translate.org.za>
+%        - Update all contact information
+%        - Add lang_lib
+%      2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%        - Added country_ab2/3, country_num
+%      2004-02-27: Dwayne Bailey <dwayne@translate.org.za>
+%        - Correction of lang_name comment	
+% 0.1 (2004-02-11):
+%	Initial Northern Sotho locale for South Africa
+%	by Zuza Software Foundation
+
+LC_IDENTIFICATION
+title      "Northern Sotho locale for South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "PO Box 28364, Sunnyside, 0132, South Africa"
+contact    "Dwayne Bailey"
+email      "dwayne@translate.org.za"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
+language   "Northern Sotho"
+territory  "South Africa"
+revision   "0.2"
+date       "2004-10-27"
+%
+category  "nso_ZA:2004";LC_IDENTIFICATION
+category  "nso_ZA:2004";LC_CTYPE
+category  "nso_ZA:2004";LC_COLLATE
+category  "nso_ZA:2004";LC_TIME
+category  "nso_ZA:2004";LC_NUMERIC
+category  "nso_ZA:2004";LC_MONETARY
+category  "nso_ZA:2004";LC_MESSAGES
+category  "nso_ZA:2004";LC_PAPER
+category  "nso_ZA:2004";LC_NAME
+category  "nso_ZA:2004";LC_ADDRESS
+category  "nso_ZA:2004";LC_TELEPHONE
+category  "nso_ZA:2004";LC_MEASUREMENT
+
+END LC_IDENTIFICATION
+
+LC_CTYPE
+% Use the characters described in the charmap file "i18n.tgz"
+copy "i18n"
+END LC_CTYPE
+
+LC_COLLATE
+% Copy the template from ISO/IEC 14651 i.e.
+% use the rules there when making ordered lists of words.
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_MONETARY
+copy "en_ZA"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "en_ZA"
+END LC_NUMERIC
+
+LC_TIME
+
+% abday - The abbreviations for the week days:
+% - Son, Mos, Bed, Rar, Ne, Hla, Mok
+abday       "<U0053><U006F><U006E>";/
+            "<U004D><U006F><U0073>";/
+            "<U0042><U0065><U0064>";/
+            "<U0052><U0061><U0072>";/
+            "<U004E><U0065>";/
+            "<U0048><U006C><U0061>";/
+            "<U004D><U006F><U006B>"
+ 
+% day - The full names of the week days:
+% - Sontaga, Mosupalogo, Labobedi, Laboraro, Labone, Labohlano, Mokibelo
+day         "<U0053><U006F><U006E><U0074><U0061><U0067><U0061>";/
+            "<U004D><U006F><U0073><U0075><U0070><U0061><U006C><U006F><U0067><U006F>";/
+            "<U004C><U0061><U0062><U006F><U0062><U0065><U0064><U0069>";/
+            "<U004C><U0061><U0062><U006F><U0072><U0061><U0072><U006F>";/
+            "<U004C><U0061><U0062><U006F><U006E><U0065>";/
+            "<U004C><U0061><U0062><U006F><U0068><U006C><U0061><U006E><U006F>";/
+            "<U004D><U006F><U006B><U0069><U0062><U0065><U006C><U006F>"
+ 
+% abmon - The abbreviations for the months 
+% - Jan, Feb, Mat, Apo, Mei, Jun, Jul, Ago, Set, Okt, Nof, Dis
+abmon       "<U004A><U0061><U006E>";"<U0046><U0065><U0062>";/
+            "<U004D><U0061><U0074>";"<U0041><U0070><U006F>";/
+            "<U004D><U0065><U0069>";"<U004A><U0075><U006E>";/
+            "<U004A><U0075><U006C>";"<U0041><U0067><U006F>";/
+            "<U0053><U0065><U0074>";"<U004F><U006B><U0074>";/
+            "<U004E><U006F><U0066>";"<U0044><U0069><U0073>"
+ 
+% mon - The full names of the months - 
+% - Janaware, Feberware, Mats^he, Aporele, Mei, June
+%   Julae, Agostose, Setemere, Oktobore, Nofemere, Disemere
+mon         "<U004A><U0061><U006E><U0061><U0077><U0061><U0072><U0065>";/
+            "<U0046><U0065><U0062><U0065><U0072><U0077><U0061><U0072><U0065>";/
+            "<U004D><U0061><U0074><U0161><U0068><U0065>";/
+            "<U0041><U0070><U006F><U0072><U0065><U006C><U0065>";/
+            "<U004D><U0065><U0069>";/
+            "<U004A><U0075><U006E><U0065>";/
+            "<U004A><U0075><U006C><U0061><U0065>";/
+            "<U0041><U0067><U006F><U0073><U0074><U006F><U0073><U0065>";/
+            "<U0053><U0065><U0074><U0065><U006D><U0065><U0072><U0065>";/
+            "<U004F><U006B><U0074><U006F><U0062><U006F><U0072><U0065>";/
+            "<U004E><U006F><U0066><U0065><U006D><U0065><U0072><U0065>";/
+            "<U0044><U0069><U0073><U0065><U006D><U0065><U0072><U0065>"
+ 
+% Abreviated date and time representation to be referenced by the "%c" field descriptor -
+d_t_fmt     "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+%
+% "%a" (short weekday name),
+% "%d" (day of month as a decimal number),
+% "%b" (short month name),
+% "%Y" (year with century as a decimal number),
+% "%T" (24-hour clock time in format HH:MM:SS),
+% "%Z" (Time zone name)
+ 
+% Date representation to be referenced by the "%x" field descriptor -
+d_fmt   "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+% "%d/%m/%Y", day/month/year as decimal numbers (01/01/2000).
+ 
+% Time representation to be referenced by the "%X" field descriptor -
+t_fmt       "<U0025><U0054>"
+% "%T" (24-hour clock time in format HH:MM:SS)
+ 
+% Define representation of ante meridiem and post meridiem strings -
+am_pm       "";""
+% The "" mean 'default to "AM" and "PM".
+ 
+% Define time representation in 12-hour format with "am_pm", to be referenced by the "%r"
+t_fmt_ampm  ""
+% The "" means that this format is not supported.
+ 
+% Date representation not described in ISO/IEC 14652. Comes out as -
+% "%a %b %e %H:%M:%S %Z %Y" which is default "date" command output
+date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
+<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U005A><U0020><U0025><U0059>"
+%
+% %a - abbreviated weekday name,
+% %b - abreviated month name,
+% %e - day of month as a decimal number with leading space (1 to 31),
+% %H - hour (24-hour clock) as a decimal number (00 to 23),
+% %M - minute as a decimal number (00 to 59),
+% %S - seconds as a decimal number (00 to 59),
+% %Z - time-zone name,
+% %Y - year with century as a decimal number,e.g. 2001.
+ 
+END LC_TIME
+
+LC_MESSAGES
+% FIXME: Check both of these
+% The affirmative response - ee
+% "^[yYeE].*"
+yesexpr     "<U005E><U005B><U0079><U0059><U0065><U0045><U005D>"
+
+% The negative response - aowa
+% "^[nNaA].*"
+noexpr      "<U005E><U005B><U006E><U004E><U0061><U0041><U005D>"
+END LC_MESSAGES
+
+LC_PAPER
+copy "en_ZA"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "en_ZA"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "en_ZA"
+END LC_MEASUREMENT
+
+LC_NAME
+% Format for addressing a person.
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+% "%d%t%g%t%m%t%f"
+% "Salutation",
+% "Empty string, or <Space>",
+% "First given name",
+% "Empty string, or <Space>",
+% "Middle names",
+% "Empty string, or <Space>",
+% "Clan names"
+ 
+% FIXME - define all the following name_*
+% General salutation for any sex
+% name_gen    ""
+
+% Salutation for unmarried females - ""
+% name_miss   ""
+ 
+% Salutation for males - ""
+% name_mr     ""
+
+% Salutation for married females - ""
+% name_mrs    ""
+
+% Salutation valid for all females - "" (no term)
+% name_ms     ""
+
+END LC_NAME
+
+LC_ADDRESS
+% Country name in Northern Sotho - "Afrika Borwa"
+country_name "<U0041><U0066><U0072><U0069><U006B><U0061><U0020>/
+<U0042><U006F><U0072><U0077><U0061>"
+ 
+% Abbreviated country postal name - "ZA"
+country_post "<U005A><U0041>"
+
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% FIXME define the following correctly
+% country_isbn ""
+ 
+% Language name in Northern Sotho - "Sesotho sa Leboa"
+lang_name "<U0053><U0065><U0073><U006F><U0074><U0068><U006F><U0020>/
+<U0073><U0061><U0020><U004C><U0065><U0062><U006F><U0061>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+lang_ab   ""
+lang_term "<U006E><U0073><U006F>"
+lang_lib  "<U006E><U0073><U006F>"
+ 
+% Representation of postal addresses (minus the addressee's name) in South
+% Africa. (Ignored for now)
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N", which gives -
+% "firm name",
+% "end of line",
+% "C/O address",
+% "end of line",
+% "department name",
+% "Building name",
+% "end of line",
+% "street or block name",
+% "space",
+% "house number or designation",
+% "space",
+% "floor number",
+% "space",
+% "room number, door designation",
+% "end of line",
+% "postal code",
+% "space",
+% "town, city",
+% "end of line",
+% "country designation for the <country_post> keyword",
+% "end of line
+%
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
+END LC_ADDRESS
+
diff -uNr libc.orig/localedata/locales/ss_ZA libc/localedata/locales/ss_ZA
--- libc.orig/localedata/locales/ss_ZA	1970-01-01 01:00:00.000000000 +0100
+++ libc/localedata/locales/ss_ZA	2004-11-09 21:17:00.000000000 +0100
@@ -0,0 +1,277 @@
+escape_char  /
+comment_char  %
+
+% ChangeLog
+% 0.2 (2004-11-09):
+%     2004-11-09 Dwayne Bailey <dwayne@translate.org.za>
+%       - Short month names comment corrected to match values in abmon
+%       - Correct yesexpr regex
+%       - <UNNNN> escaped all entries
+%     2004-11-08 Dwayne Bailey <dwayne@translate.org.za>
+%       - Changed all contact details
+%       - Change %d to %-e for d_t_fmt and date_fmt
+%       - Remove .* from yes/noexpr
+%       - <UNNNN> escaped some entries
+%     2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%       - Added country_ab2/3, country_num
+%     2004-02-27 Dwayne Bailey <dwayne@translate.org.za>
+%       - Correct capatilisation of lang_name
+% 0.1 (2004-02-24):
+%	    - Initial Swati locale for South Africa
+%         by Zuza Software Foundation
+
+LC_IDENTIFICATION
+title      "Swati locale for South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "PO Box 28364, Sunnyside, 0132, South Africa"
+contact    "Dwayne Bailey"
+email      "dwayne@translate.org.za"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
+language   "Swati"
+territory  "South Africa"
+revision   "0.2"
+date       "2004-11-08"
+%
+category  "ss_ZA:2004";LC_IDENTIFICATION
+category  "ss_ZA:2004";LC_CTYPE
+category  "ss_ZA:2004";LC_COLLATE
+category  "ss_ZA:2004";LC_TIME
+category  "ss_ZA:2004";LC_NUMERIC
+category  "ss_ZA:2004";LC_MONETARY
+category  "ss_ZA:2004";LC_MESSAGES
+category  "ss_ZA:2004";LC_PAPER
+category  "ss_ZA:2004";LC_NAME
+category  "ss_ZA:2004";LC_ADDRESS
+category  "ss_ZA:2004";LC_TELEPHONE
+category  "ss_ZA:2004";LC_MEASUREMENT
+
+END LC_IDENTIFICATION
+
+LC_CTYPE
+% Use the characters described in the charmap file "i18n.tgz"
+copy "i18n"
+END LC_CTYPE
+
+LC_COLLATE
+% Copy the template from ISO/IEC 14651 i.e.
+% use the rules there when making ordered lists of words.
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_MONETARY
+copy "en_ZA"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "en_ZA"
+END LC_NUMERIC
+
+LC_TIME
+
+% abday - The abbreviations for the week days:
+% - Son, Mso, Bil, Tsa, Ne, Hla, Mgc
+abday       "<U0053><U006F><U006E>";/
+            "<U004D><U0073><U006F>";/
+            "<U0042><U0069><U006C>";/
+            "<U0054><U0073><U0061>";/
+            "<U004E><U0065>";/
+            "<U0048><U006C><U0061>";/
+            "<U004D><U0067><U0063>"
+ 
+% day - The full names of the week days:
+% - Lisontfo, uMsombuluko, Lesibili, Lesitsatfu, Lesine, Lesihlanu, uMgcibelo
+day         "<U004C><U0069><U0073><U006F><U006E><U0074><U0066><U006F>";/
+            "<U0075><U004D><U0073><U006F><U006D><U0062><U0075><U006C><U0075><U006B><U006F>";/
+            "<U004C><U0065><U0073><U0069><U0062><U0069><U006C><U0069>";/
+            "<U004C><U0065><U0073><U0069><U0074><U0073><U0061><U0074><U0066><U0075>";/
+            "<U004C><U0065><U0073><U0069><U006E><U0065>";/
+            "<U004C><U0065><U0073><U0069><U0068><U006C><U0061><U006E><U0075>";/
+            "<U0075><U004D><U0067><U0063><U0069><U0062><U0065><U006C><U006F>"
+ 
+% abmon - The abbreviations for the months 
+% - Bhi, Van, Vul, Mab, Nkh, Nhl, Kho, Ngc, Nyo, Mph, Lwe, Ngo
+abmon       "<U0042><U0068><U0069>";"<U0056><U0061><U006E>";/
+            "<U0056><U0075><U006C>";"<U004D><U0061><U0062>";/
+            "<U004E><U006B><U0068>";"<U004E><U0068><U006C>";/
+            "<U004B><U0068><U006F>";"<U004E><U0067><U0063>";/
+            "<U004E><U0079><U006F>";"<U004D><U0070><U0068>";/
+            "<U004C><U0077><U0065>";"<U004E><U0067><U006F>"
+ 
+% mon - The full names of the months - 
+% - Bhimbidvwane, iNdlovana, iNdlovu-lenkhulu, Mabasa, iNkhwekhweti, iNhlaba
+%   Kholwane, iNgci, iNyoni, iMphala, Lweti, iNgongoni
+mon         "<U0042><U0068><U0069><U006D><U0062><U0069><U0064><U0076><U0077><U0061><U006E><U0065>";/
+            "<U0069><U004E><U0064><U006C><U006F><U0076><U0061><U006E><U0061>";/
+            "<U0069><U004E><U0064><U006C><U006F><U0076><U0075><U002D><U006C><U0065><U006E><U006B><U0068><U0075><U006C><U0075>";/
+            "<U004D><U0061><U0062><U0061><U0073><U0061>";/
+            "<U0069><U004E><U006B><U0068><U0077><U0065><U006B><U0068><U0077><U0065><U0074><U0069>";/
+            "<U0069><U004E><U0068><U006C><U0061><U0062><U0061>";/
+            "<U004B><U0068><U006F><U006C><U0077><U0061><U006E><U0065>";/
+            "<U0069><U004E><U0067><U0063><U0069>";/
+            "<U0069><U004E><U0079><U006F><U006E><U0069>";/
+            "<U0069><U004D><U0070><U0068><U0061><U006C><U0061>";/
+            "<U004C><U0077><U0065><U0074><U0069>";/
+            "<U0069><U004E><U0067><U006F><U006E><U0067><U006F><U006E><U0069>"
+ 
+% Abreviated date and time representation to be referenced by the "%c" field descriptor -
+d_t_fmt     "<U0025><U0061><U0020><U0025><U002D><U0065><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+%
+% "%a" (short weekday name),
+% "%-e" (day of month as a decimal number),
+% "%b" (short month name),
+% "%Y" (year with century as a decimal number),
+% "%T" (24-hour clock time in format HH:MM:SS),
+% "%Z" (Time zone name)
+ 
+% Date representation to be referenced by the "%x" field descriptor -
+d_fmt   "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+% "%d/%m/%Y", day/month/year as decimal numbers (01/01/2000).
+ 
+% Time representation to be referenced by the "%X" field descriptor -
+t_fmt       "<U0025><U0054>"
+% "%T" (24-hour clock time in format HH:MM:SS)
+ 
+% Define representation of ante meridiem and post meridiem strings -
+am_pm       "";""
+% The "" mean 'default to "AM" and "PM".
+ 
+% Define time representation in 12-hour format with "am_pm", to be referenced by the "%r"
+t_fmt_ampm  ""
+% The "" means that this format is not supported.
+ 
+% Date representation not described in ISO/IEC 14652. Comes out as -
+% "%a %b %-e %H:%M:%S %Z %Y" which is default "date" command output
+date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U002D><U0065>/
+<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U005A><U0020><U0025><U0059>"
+%
+% %a - abbreviated weekday name,
+% %b - abreviated month name,
+% %-e - day of month as a decimal number without leading space (1 to 31),
+% %H - hour (24-hour clock) as a decimal number (00 to 23),
+% %M - minute as a decimal number (00 to 59),
+% %S - seconds as a decimal number (00 to 59),
+% %Z - time-zone name,
+% %Y - year with century as a decimal number,e.g. 2001.
+ 
+END LC_TIME
+
+LC_MESSAGES
+% FIXME: Check both of these
+% The affirmative response - yebo
+% "^[yY]"
+yesexpr     "<U005E><U005B><U0079><U0059><U005D>"
+
+% The negative response - akukho
+% "^[nNaA]"
+noexpr      "<U005E><U005B><U006E><U004E><U0061><U0041><U005D>"
+END LC_MESSAGES
+
+LC_PAPER
+copy "en_ZA"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "en_ZA"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "en_ZA"
+END LC_MEASUREMENT
+
+LC_NAME
+% Format for addressing a person.
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+% "%d%t%g%t%m%t%f"
+% "Salutation",
+% "Empty string, or <Space>",
+% "First given name",
+% "Empty string, or <Space>",
+% "Middle names",
+% "Empty string, or <Space>",
+% "Clan names"
+ 
+% FIXME - define all the following name_*
+% General salutation for any sex
+% name_gen    ""
+
+% Salutation for unmarried females - ""
+% name_miss   ""
+ 
+% Salutation for males - ""
+% name_mr     ""
+
+% Salutation for married females - ""
+% name_mrs    ""
+
+% Salutation valid for all females - "" (no term)
+% name_ms     ""
+
+END LC_NAME
+
+LC_ADDRESS
+% Country name in Swati - "iNingizimu Afrika"
+country_name "<U0069><U004E><U0069><U006E><U0067><U0069><U007A><U0069><U006D><U0075><U0020><U0041><U0066><U0072><U0069><U006B><U0061>"
+ 
+% Abbreviated country postal name - "ZA"
+country_post "<U005A><U0041>"
+ 
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% FIXME define the following correctly
+% country_isbn ""
+ 
+% Language name in Swati - "SiSwati"
+lang_name "<U0053><U0069><U0053><U0077><U0061><U0074><U0069>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+% "ss", ssw"
+lang_ab   "<U0073><U0073>"
+lang_term "<U0073><U0073><U0077>"
+ 
+% Representation of postal addresses (minus the addressee's name) in South
+% Africa. (Ignored for now)
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N", which gives -
+% "firm name",
+% "end of line",
+% "C/O address",
+% "end of line",
+% "department name",
+% "Building name",
+% "end of line",
+% "street or block name",
+% "space",
+% "house number or designation",
+% "space",
+% "floor number",
+% "space",
+% "room number, door designation",
+% "end of line",
+% "postal code",
+% "space",
+% "town, city",
+% "end of line",
+% "country designation for the <country_post> keyword",
+% "end of line
+%
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
+END LC_ADDRESS
+
diff -uNr libc.orig/localedata/locales/st_ZA libc/localedata/locales/st_ZA
--- libc.orig/localedata/locales/st_ZA	2003-12-05 01:26:00.000000000 +0100
+++ libc/localedata/locales/st_ZA	2004-11-08 17:26:30.000000000 +0100
@@ -2,22 +2,32 @@
 comment_char  %
 
 % ChangeLog
+% 0.2 (2004-11-03):
+%    2004-11-02 Dwayne Bailey <dwayne@translate.org.za>
+%      - Change all contact information
+%      - Change %e in date_fmt and d_t_fmt to %-e
+%      - Remove .* from yes/noexpr
+%    2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%      - Added country_ab2/3, country_num
+%      - Added country_car
+%    2004-03-29 Dwayne Bailey <dwayne@translate.org.za>
+%      - Added lang_ab and lang_term
 % 0.1 (2003-04-21):
 %   - Initial Sotho locale for South Africa
 %     by Zuza Software Foundation
 
 LC_IDENTIFICATION
 title      "Sotho locale for South Africa"
-source     "Zuza Software Foundation"
-address    "PO Box 13412, Mowbray, 7705, South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "PO Box 28364, Sunnyside, 0132, South Africa"
 contact    "Dwayne Bailey"
 email      "dwayne@translate.org.za"
-tel        "+27 (0)21 448 9265"
-fax        "+27 (0)21 448 9574"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
 language   "Sotho"
 territory  "South Africa"
-revision   "0.1"
-date       "2003-04-21"
+revision   "0.2"
+date       "2004-11-03"
 %
 category  "st_ZA:2003";LC_IDENTIFICATION
 category  "st_ZA:2003";LC_CTYPE
@@ -101,10 +111,11 @@
             "<U0054><U0073><U0068><U0069><U0074><U0077><U0065>"
  
 % Abreviated date and time representation to be referenced by the "%c" field descriptor -
-d_t_fmt     "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
-%
+d_t_fmt     "<U0025><U0061><U0020><U0025><U002D><U0065><U0020><U0025><U0062>/
+<U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+
 % "%a" (short weekday name),
-% "%d" (day of month as a decimal number),
+% "%-e" (day of month as a decimal number),
 % "%b" (short month name),
 % "%Y" (year with century as a decimal number),
 % "%T" (24-hour clock time in format HH:MM:SS),
@@ -127,14 +138,14 @@
 % The "" means that this format is not supported.
  
 % Date representation not described in ISO/IEC 14652. Comes out as -
-% "%a %b %e %H:%M:%S %Z %Y" which is default "date" command output
-date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
+% "%a %b %-e %H:%M:%S %Z %Y" which is default "date" command output
+date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U002D><U0065>/
 <U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
 <U0025><U005A><U0020><U0025><U0059>"
 %
 % %a - abbreviated weekday name,
 % %b - abreviated month name,
-% %e - day of month as a decimal number with leading space (1 to 31),
+% %-e - day of month as a decimal number without leading space (1 to 31),
 % %H - hour (24-hour clock) as a decimal number (00 to 23),
 % %M - minute as a decimal number (00 to 59),
 % %S - seconds as a decimal number (00 to 59),
@@ -146,12 +157,12 @@
 LC_MESSAGES
 % FIXME: Check both of these
 % The affirmative response -
-% "^[yY].*"
-yesexpr     "<U005E><U005B><U0079><U0059><U005D><U002E><U002A>"
+% "^[yY]"
+yesexpr     "<U005E><U005B><U0079><U0059><U005D>"
 
 % The negative response -
-% "^[nN].*"
-noexpr      "<U005E><U005B><U006E><U004E><U005D><U002E><U002A>"
+% "^[nN]"
+noexpr      "<U005E><U005B><U006E><U004E><U005D>"
 END LC_MESSAGES
 
 LC_PAPER
@@ -204,21 +215,22 @@
 % Abbreviated country postal name - "ZA"
 country_post "<U005A><U0041>"
  
-% ISO 3166 two letter country abbreviation - "ZA"
-country_ab2 "<U005A><U0041>"
-
 % FIXME define the following correctly
-% country_ab3 ""
-% country_car ""
 % country_isbn ""
-% country_num 
  
 % Language name in Sotho - "Sesotho"
 lang_name "<U0053><U0065><U0073><U006F><U0074><U0068><U006F>"
 
-% FIXME define the following correctly
-% lang_ab   ""
-% lang_term ""
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+% "st", "sot"
+lang_ab   "<U0073><U0074>"
+lang_term "<U0073><U006F><U0074>"
  
 % Representation of postal addresses (minus the addressee's name) in South
 % Africa. (Ignored for now)
@@ -250,6 +262,13 @@
 % "country designation for the <country_post> keyword",
 % "end of line
 %
-country_num 710
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
 END LC_ADDRESS
 
diff -uNr libc.orig/localedata/locales/tn_ZA libc/localedata/locales/tn_ZA
--- libc.orig/localedata/locales/tn_ZA	1970-01-01 01:00:00.000000000 +0100
+++ libc/localedata/locales/tn_ZA	2004-11-08 19:50:02.000000000 +0100
@@ -0,0 +1,276 @@
+escape_char  /
+comment_char  %
+
+% ChangeLog
+% 0.2 (2004-11-08):
+%     2004-11-08 Dwayne Bailey <dwayne@translate.org.za>
+%       - Changed all contact information
+%       - d_t_fmt, date_fmt changed %d to %-e
+%       - Remove .* from yes/noexpr
+%       - Convert all to <UNNNN> syntax
+%     2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%       - Added country_ab2/3, country_num
+%     2004-02-27 Dwayne Bailey <dwayne@translate.org.za>
+%       - Correct capatilisation of lang_name
+%       - Validated and corrected bad month names
+% 0.1 (2004-02-24):
+%	  - Initial Tswana locale for South Africa
+%	    by Zuza Software Foundation
+
+LC_IDENTIFICATION
+title      "Tswana locale for South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "PO Box 28364, Sunnyside, 0132, South Africa"
+contact    "Dwayne Bailey"
+email      "dwayne@translate.org.za"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
+language   "Tswana"
+territory  "South Africa"
+revision   "0.2"
+date       "2004-11-08"
+%
+category  "tn_ZA:2004";LC_IDENTIFICATION
+category  "tn_ZA:2004";LC_CTYPE
+category  "tn_ZA:2004";LC_COLLATE
+category  "tn_ZA:2004";LC_TIME
+category  "tn_ZA:2004";LC_NUMERIC
+category  "tn_ZA:2004";LC_MONETARY
+category  "tn_ZA:2004";LC_MESSAGES
+category  "tn_ZA:2004";LC_PAPER
+category  "tn_ZA:2004";LC_NAME
+category  "tn_ZA:2004";LC_ADDRESS
+category  "tn_ZA:2004";LC_TELEPHONE
+category  "tn_ZA:2004";LC_MEASUREMENT
+
+END LC_IDENTIFICATION
+
+LC_CTYPE
+% Use the characters described in the charmap file "i18n.tgz"
+copy "i18n"
+END LC_CTYPE
+
+LC_COLLATE
+% Copy the template from ISO/IEC 14651 i.e.
+% use the rules there when making ordered lists of words.
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_MONETARY
+copy "en_ZA"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "en_ZA"
+END LC_NUMERIC
+
+LC_TIME
+
+% abday - The abbreviations for the week days:
+% - Tsh, Mos, Bed, Rar, Ne, Tlh, Mat
+abday       "<U0054><U0073><U0068>";/
+            "<U004D><U006F><U0073>";/
+            "<U0042><U0065><U0064>";/
+            "<U0052><U0061><U0072>";/
+            "<U004E><U0065>";/
+            "<U0054><U006C><U0068>";/
+            "<U004D><U0061><U0074>"
+ 
+% day - The full names of the week days:
+% - Tshipi, Mosopulogo, Labobedi, Laboraro, Labone, Labotlhano, Matlhatso
+day         "<U0054><U0073><U0068><U0069><U0070><U0069>";/
+            "<U004D><U006F><U0073><U006F><U0070><U0075><U006C><U006F><U0067><U006F>";/
+            "<U004C><U0061><U0062><U006F><U0062><U0065><U0064><U0069>";/
+            "<U004C><U0061><U0062><U006F><U0072><U0061><U0072><U006F>";/
+            "<U004C><U0061><U0062><U006F><U006E><U0065>";/
+            "<U004C><U0061><U0062><U006F><U0074><U006C><U0068><U0061><U006E><U006F>";/
+            "<U004D><U0061><U0074><U006C><U0068><U0061><U0074><U0073><U006F>"
+ 
+% abmon - The abbreviations for the months 
+% - Hir, Tlh, Mop, Mor, Mot, See, Phu, Pha, Lwe, Lan, Ngw, Mar
+abmon       "<U0048><U0069><U0072>";"<U0054><U006C><U0068>";/
+            "<U004D><U006F><U0070>";"<U004D><U006F><U0072>";/
+            "<U004D><U006F><U0074>";"<U0053><U0065><U0065>";/
+            "<U0050><U0068><U0075>";"<U0050><U0068><U0061>";/
+            "<U004C><U0077><U0065>";"<U004C><U0061><U006E>";/
+            "<U004E><U0067><U0077>";"<U004D><U0061><U0072>"
+ 
+% mon - The full names of the months - 
+% - Ferikgong, Tlhakole, Mopitlo, Moranang, Motsheganang, Seetebosigo
+%   Phukwi, Phatwe, Lwetse, Diphalane, Ngwanatsele, Sedimonthole
+mon         "<U0046><U0065><U0072><U0069><U006B><U0067><U006F><U006E><U0067>";/
+            "<U0054><U006C><U0068><U0061><U006B><U006F><U006C><U0065>";/
+            "<U004D><U006F><U0070><U0069><U0074><U006C><U006F>";/
+            "<U004D><U006F><U0072><U0061><U006E><U0061><U006E><U0067>";/
+            "<U004D><U006F><U0074><U0073><U0068><U0065><U0067><U0061><U006E><U0061><U006E><U0067>";/
+            "<U0053><U0065><U0065><U0074><U0065><U0062><U006F><U0073><U0069><U0067><U006F>";/
+            "<U0050><U0068><U0075><U006B><U0077><U0069>";/
+            "<U0050><U0068><U0061><U0074><U0077><U0065>";/
+            "<U004C><U0077><U0065><U0074><U0073><U0065>";/
+            "<U0044><U0069><U0070><U0068><U0061><U006C><U0061><U006E><U0065>";/
+            "<U004E><U0067><U0077><U0061><U006E><U0061><U0074><U0073><U0065><U006C><U0065>";/
+            "<U0053><U0065><U0064><U0069><U006D><U006F><U006E><U0074><U0068><U006F><U006C><U0065>"
+
+% Abreviated date and time representation to be referenced by the "%c" field descriptor -
+d_t_fmt     "<U0025><U0061><U0020><U0025><U002D><U0065><U0020><U0025><U0062><U0020>/
+<U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+%
+% "%a" (short weekday name),
+% "%-e" (day of month as a decimal number),
+% "%b" (short month name),
+% "%Y" (year with century as a decimal number),
+% "%T" (24-hour clock time in format HH:MM:SS),
+% "%Z" (Time zone name)
+ 
+% Date representation to be referenced by the "%x" field descriptor -
+d_fmt   "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+% "%d/%m/%Y", day/month/year as decimal numbers (01/01/2000).
+ 
+% Time representation to be referenced by the "%X" field descriptor -
+t_fmt       "<U0025><U0054>"
+% "%T" (24-hour clock time in format HH:MM:SS)
+ 
+% Define representation of ante meridiem and post meridiem strings -
+am_pm       "";""
+% The "" mean 'default to "AM" and "PM".
+ 
+% Define time representation in 12-hour format with "am_pm", to be referenced by the "%r"
+t_fmt_ampm  ""
+% The "" means that this format is not supported.
+ 
+% Date representation not described in ISO/IEC 14652. Comes out as -
+% "%a %b %-e %H:%M:%S %Z %Y" which is default "date" command output
+date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U002D><U0065>/
+<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U005A><U0020><U0025><U0059>"
+%
+% %a - abbreviated weekday name,
+% %b - abreviated month name,
+% %-e - day of month as a decimal number with leading space (1 to 31),
+% %H - hour (24-hour clock) as a decimal number (00 to 23),
+% %M - minute as a decimal number (00 to 59),
+% %S - seconds as a decimal number (00 to 59),
+% %Z - time-zone name,
+% %Y - year with century as a decimal number,e.g. 2001.
+ 
+END LC_TIME
+
+LC_MESSAGES
+% FIXME: Check both of these
+% The affirmative response - ee
+% "^[yYeE]"
+yesexpr     "<U005E><U005B><U0079><U0059><U0065><U0045><U005D>"
+
+% The negative response - nyaa
+% "^[nN]"
+noexpr      "<U005E><U005B><U006E><U004E><U005D>"
+END LC_MESSAGES
+
+LC_PAPER
+copy "en_ZA"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "en_ZA"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "en_ZA"
+END LC_MEASUREMENT
+
+LC_NAME
+% Format for addressing a person.
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+% "%d%t%g%t%m%t%f"
+% "Salutation",
+% "Empty string, or <Space>",
+% "First given name",
+% "Empty string, or <Space>",
+% "Middle names",
+% "Empty string, or <Space>",
+% "Clan names"
+ 
+% FIXME - define all the following name_*
+% General salutation for any sex
+% name_gen    ""
+
+% Salutation for unmarried females - ""
+% name_miss   ""
+ 
+% Salutation for males - ""
+% name_mr     ""
+
+% Salutation for married females - ""
+% name_mrs    ""
+
+% Salutation valid for all females - "" (no term)
+% name_ms     ""
+
+END LC_NAME
+
+LC_ADDRESS
+% Country name in Tswana - "Aforika Borwa"
+country_name "<U0041><U0066><U006F><U0072><U0069><U006B><U0061><U0020>/
+<U0042><U006F><U0072><U0077><U0061>"
+ 
+% Abbreviated country postal name - "ZA"
+country_post "<U005A><U0041>"
+ 
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% FIXME define the following correctly
+% country_isbn ""
+ 
+% Language name in Tswana - "Setswana"
+lang_name "<U0053><U0065><U0074><U0073><U0077><U0061><U006E><U0061>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+% "tn", "tsn"
+lang_ab   "<U0074><U006E>"
+lang_term "<U0074><U0073><U006E>"
+ 
+% Representation of postal addresses (minus the addressee's name) in South
+% Africa. (Ignored for now)
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N", which gives -
+% "firm name",
+% "end of line",
+% "C/O address",
+% "end of line",
+% "department name",
+% "Building name",
+% "end of line",
+% "street or block name",
+% "space",
+% "house number or designation",
+% "space",
+% "floor number",
+% "space",
+% "room number, door designation",
+% "end of line",
+% "postal code",
+% "space",
+% "town, city",
+% "end of line",
+% "country designation for the <country_post> keyword",
+% "end of line
+%
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
+END LC_ADDRESS
+
diff -uNr libc.orig/localedata/locales/ts_ZA libc/localedata/locales/ts_ZA
--- libc.orig/localedata/locales/ts_ZA	1970-01-01 01:00:00.000000000 +0100
+++ libc/localedata/locales/ts_ZA	2004-11-09 20:34:30.000000000 +0100
@@ -0,0 +1,274 @@
+escape_char  /
+comment_char  %
+
+% ChangeLog
+% 0.2 (2004-11-09):
+%     2004-11-09 Dwayne Bailey <dwayne@translate.org.za>
+%       - Change %d to %-e for d_t_fmt and date_fmt
+%       - <UNNNN> escape everything
+%       - Changed all contact information
+%       - Removed .* from yes/noexpr
+%     2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%       - Added country_ab2/3, country_num
+%     2004-02-27 Dwayne Bailey <dwayne@translate.org.za>
+%	    - Correct capatilisation of lang_name
+% 0.1 (2004-02-24):
+%	   - Initial Tsonga locale for South Africa
+%	     by Zuza Software Foundation
+
+LC_IDENTIFICATION
+title      "Tsonga locale for South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "PO Box 28364, Sunnyside, 0132, South Africa"
+contact    "Dwayne Bailey"
+email      "dwayne@translate.org.za"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
+language   "Tsonga"
+territory  "South Africa"
+revision   "0.2"
+date       "2004-11-09"
+%
+category  "ts_ZA:2004";LC_IDENTIFICATION
+category  "ts_ZA:2004";LC_CTYPE
+category  "ts_ZA:2004";LC_COLLATE
+category  "ts_ZA:2004";LC_TIME
+category  "ts_ZA:2004";LC_NUMERIC
+category  "ts_ZA:2004";LC_MONETARY
+category  "ts_ZA:2004";LC_MESSAGES
+category  "ts_ZA:2004";LC_PAPER
+category  "ts_ZA:2004";LC_NAME
+category  "ts_ZA:2004";LC_ADDRESS
+category  "ts_ZA:2004";LC_TELEPHONE
+category  "ts_ZA:2004";LC_MEASUREMENT
+
+END LC_IDENTIFICATION
+
+LC_CTYPE
+% Use the characters described in the charmap file "i18n.tgz"
+copy "i18n"
+END LC_CTYPE
+
+LC_COLLATE
+% Copy the template from ISO/IEC 14651 i.e.
+% use the rules there when making ordered lists of words.
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_MONETARY
+copy "en_ZA"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "en_ZA"
+END LC_NUMERIC
+
+LC_TIME
+
+% abday - The abbreviations for the week days:
+% - Son, Mus, Bir, Har, Ne, Tlh, Muq
+abday       "<U0053><U006F><U006E>";/
+            "<U004D><U0075><U0073>";/
+            "<U0042><U0069><U0072>";/
+            "<U0048><U0061><U0072>";/
+            "<U004E><U0065>";/
+            "<U0054><U006C><U0068>";/
+            "<U004D><U0075><U0071>"
+ 
+% day - The full names of the week days:
+% - Sonto, Musumbunuku, Ravumbirhi, Ravurharhu, Ravumune, Ravuntlhanu, Muqivela
+day         "<U0053><U006F><U006E><U0074><U006F>";/
+            "<U004D><U0075><U0073><U0075><U006D><U0062><U0075><U006E><U0075><U006B><U0075>";/
+            "<U0052><U0061><U0076><U0075><U006D><U0062><U0069><U0072><U0068><U0069>";/
+            "<U0052><U0061><U0076><U0075><U0072><U0068><U0061><U0072><U0068><U0075>";/
+            "<U0052><U0061><U0076><U0075><U006D><U0075><U006E><U0065>";/
+            "<U0052><U0061><U0076><U0075><U006E><U0074><U006C><U0068><U0061><U006E><U0075>";/
+            "<U004D><U0075><U0071><U0069><U0076><U0065><U006C><U0061>"
+            
+ 
+% abmon - The abbreviations for the months 
+% - Jan, Feb, Mac, Api, Mey, Jun, Jul, Ago, Sep, Okt, Nov, Dis
+abmon       "<U004A><U0061><U006E>";"<U0046><U0065><U0062>";/
+            "<U004D><U0061><U0063>";"<U0041><U0070><U0069>";/
+            "<U004D><U0065><U0079>";"<U004A><U0075><U006E>";/
+            "<U004A><U0075><U006C>";"<U0041><U0067><U006F>";/
+            "<U0053><U0065><U0070>";"<U004F><U006B><U0074>";/
+            "<U004E><U006F><U0076>";"<U0044><U0069><U0073>"
+ 
+% mon - The full names of the months - 
+% - Janiwari, Febriwari, Machi, Apireli, Meyi, Juni
+%   Julayi, Agoste, Septembere, Oktoba, Novhemba, Disemba
+mon         "<U004A><U0061><U006E><U0069><U0077><U0061><U0072><U0069>";/
+            "<U0046><U0065><U0062><U0072><U0069><U0077><U0061><U0072><U0069>";/
+            "<U004D><U0061><U0063><U0068><U0069>";/
+            "<U0041><U0070><U0069><U0072><U0065><U006C><U0069>";/
+            "<U004D><U0065><U0079><U0069>";/
+            "<U004A><U0075><U006E><U0069>";/
+            "<U004A><U0075><U006C><U0061><U0079><U0069>";/
+            "<U0041><U0067><U006F><U0073><U0074><U0065>";/
+            "<U0053><U0065><U0070><U0074><U0065><U006D><U0062><U0065><U0072><U0065>";/
+            "<U004F><U006B><U0074><U006F><U0062><U0061>";/
+            "<U004E><U006F><U0076><U0068><U0065><U006D><U0062><U0061>";/
+            "<U0044><U0069><U0073><U0065><U006D><U0062><U0061>"
+
+% Abreviated date and time representation to be referenced by the "%c" field descriptor -
+d_t_fmt     "<U0025><U0061><U0020><U0025><U002D><U0065><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+%
+% "%a" (short weekday name),
+% "%-e" (day of month as a decimal number),
+% "%b" (short month name),
+% "%Y" (year with century as a decimal number),
+% "%T" (24-hour clock time in format HH:MM:SS),
+% "%Z" (Time zone name)
+ 
+% Date representation to be referenced by the "%x" field descriptor -
+d_fmt   "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+% "%d/%m/%Y", day/month/year as decimal numbers (01/01/2000).
+ 
+% Time representation to be referenced by the "%X" field descriptor -
+t_fmt       "<U0025><U0054>"
+% "%T" (24-hour clock time in format HH:MM:SS)
+ 
+% Define representation of ante meridiem and post meridiem strings -
+am_pm       "";""
+% The "" mean 'default to "AM" and "PM".
+ 
+% Define time representation in 12-hour format with "am_pm", to be referenced by the "%r"
+t_fmt_ampm  ""
+% The "" means that this format is not supported.
+ 
+% Date representation not described in ISO/IEC 14652. Comes out as -
+% "%a %b %-e %H:%M:%S %Z %Y" which is default "date" command output
+date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U002D><U0065>/
+<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U005A><U0020><U0025><U0059>"
+%
+% %a - abbreviated weekday name,
+% %b - abreviated month name,
+% %-e - day of month as a decimal number without leading space (1 to 31),
+% %H - hour (24-hour clock) as a decimal number (00 to 23),
+% %M - minute as a decimal number (00 to 59),
+% %S - seconds as a decimal number (00 to 59),
+% %Z - time-zone name,
+% %Y - year with century as a decimal number,e.g. 2001.
+ 
+END LC_TIME
+
+LC_MESSAGES
+% FIXME: Check both of these
+% The affirmative response - TODO
+% "^[yY]"
+yesexpr     "<U005E><U005B><U0079><U0059><U005D>"
+
+% The negative response - TODO
+% "^[nN]"
+noexpr      "<U005E><U005B><U006E><U004E><U005D>"
+END LC_MESSAGES
+
+LC_PAPER
+copy "en_ZA"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "en_ZA"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "en_ZA"
+END LC_MEASUREMENT
+
+LC_NAME
+% Format for addressing a person.
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+% "%d%t%g%t%m%t%f"
+% "Salutation",
+% "Empty string, or <Space>",
+% "First given name",
+% "Empty string, or <Space>",
+% "Middle names",
+% "Empty string, or <Space>",
+% "Clan names"
+ 
+% FIXME - define all the following name_*
+% General salutation for any sex
+% name_gen    ""
+
+% Salutation for unmarried females - ""
+% name_miss   ""
+ 
+% Salutation for males - ""
+% name_mr     ""
+
+% Salutation for married females - ""
+% name_mrs    ""
+
+% Salutation valid for all females - "" (no term)
+% name_ms     ""
+
+END LC_NAME
+
+LC_ADDRESS
+% Country name in Tsonga - "Afrika Dzonga"
+country_name "<U0041><U0066><U0072><U0069><U006B><U0061><U0020><U0044><U007A><U006F><U006E><U0067><U0061>"
+ 
+% Abbreviated country postal name - "ZA"
+country_post "<U005A><U0041>"
+ 
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% FIXME define the following correctly
+% country_isbn ""
+ 
+% Language name in Tsonga - "Xitsonga"
+lang_name "<U0058><U0069><U0074><U0073><U006F><U006E><U0067><U0061>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+% "ts", "tso"
+lang_ab   "<U0074><U0073>"
+lang_term "<U0074><U0073><U006F>"
+ 
+% Representation of postal addresses (minus the addressee's name) in South
+% Africa. (Ignored for now)
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N", which gives -
+% "firm name",
+% "end of line",
+% "C/O address",
+% "end of line",
+% "department name",
+% "Building name",
+% "end of line",
+% "street or block name",
+% "space",
+% "house number or designation",
+% "space",
+% "floor number",
+% "space",
+% "room number, door designation",
+% "end of line",
+% "postal code",
+% "space",
+% "town, city",
+% "end of line",
+% "country designation for the <country_post> keyword",
+% "end of line
+%
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
+END LC_ADDRESS
+
diff -uNr libc.orig/localedata/locales/ve_ZA libc/localedata/locales/ve_ZA
--- libc.orig/localedata/locales/ve_ZA	1970-01-01 01:00:00.000000000 +0100
+++ libc/localedata/locales/ve_ZA	2004-10-29 17:16:19.000000000 +0200
@@ -0,0 +1,280 @@
+escape_char  /
+comment_char  %
+
+% ChangeLog
+% 0.2 (2004-10-29):
+%      2004-10-29 Dwayne Bailey <dwayne@translate.org.za>
+%        - Converted to <UNNNN> syntax for everything
+%        - Correct yes/noexpr
+%        - Corrected some days of the week
+%      2004-10-28 Dwayne Bailey <dwayne@translate.org.za>
+%        - Updated all contact information
+%        - Added Translate.org.za to "source" for clarity
+%      2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%        - Added country_ab2/3, country_num
+%      2004-02-27 Dwayne Bailey <dwayne@translate.org.za>
+%	     - Correct capatilisation of lang_name
+% 0.1 (2004-02-24):
+%	   2004-02-24 Dwayne Bailey <dwayne@translate.org.za>
+%	     - Initial Venda locale for South Africa
+%          by Zuza Software Foundation
+
+LC_IDENTIFICATION
+title      "Venda locale for South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "PO Box 28364, Sunnyside, 0132, South Africa"
+contact    "Dwayne Bailey"
+email      "dwayne@translate.org.za"
+tel        "+27 12 343 0389"
+fax        "+27 12 343 0389"
+language   "Venda"
+territory  "South Africa"
+revision   "0.2"
+date       "2004-10-29"
+%
+category  "ve_ZA:2004";LC_IDENTIFICATION
+category  "ve_ZA:2004";LC_CTYPE
+category  "ve_ZA:2004";LC_COLLATE
+category  "ve_ZA:2004";LC_TIME
+category  "ve_ZA:2004";LC_NUMERIC
+category  "ve_ZA:2004";LC_MONETARY
+category  "ve_ZA:2004";LC_MESSAGES
+category  "ve_ZA:2004";LC_PAPER
+category  "ve_ZA:2004";LC_NAME
+category  "ve_ZA:2004";LC_ADDRESS
+category  "ve_ZA:2004";LC_TELEPHONE
+category  "ve_ZA:2004";LC_MEASUREMENT
+
+END LC_IDENTIFICATION
+
+LC_CTYPE
+% Use the characters described in the charmap file "i18n.tgz"
+copy "i18n"
+END LC_CTYPE
+
+LC_COLLATE
+% Copy the template from ISO/IEC 14651 i.e.
+% use the rules there when making ordered lists of words.
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_MONETARY
+copy "en_ZA"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "en_ZA"
+END LC_NUMERIC
+
+LC_TIME
+
+% abday - The abbreviations for the week days:
+% - Swo, Mus, Vhi, Rar, L^a, T^an, Mug
+abday       "<U0053><U0077><U006F>";/
+            "<U004D><U0075><U0073>";/
+            "<U0056><U0068><U0069>";/
+            "<U0052><U0061><U0072>";/
+            "<U1E4B><U0061>";/
+            "<U1E70><U0061><U006E>";/
+            "<U004D><U0075><U0067>"
+ 
+% day - The full names of the week days:
+% - Swondaha, Musumbuluwo, L^avhuvhili, L^avhuraru, L^avhun^a, L^avhut^anu, Mugivhela
+day         "<U0053><U0077><U006F><U006E><U0064><U0061><U0068><U0061>";/
+            "<U004D><U0075><U0073><U0075><U006D><U0062><U0075><U006C><U0075><U0077><U006F>";/
+            "<U1E3C><U0061><U0076><U0068><U0075><U0076><U0068><U0069><U006C><U0069>";/
+            "<U1E3C><U0061><U0076><U0068><U0075><U0072><U0061><U0072><U0075>";/
+            "<U1E3C><U0061><U0076><U0068><U0075><U1E4B><U0061>";/
+            "<U1E3C><U0061><U0076><U0068><U0075><U1E71><U0061><U006E><U0075>";/
+            "<U004D><U0075><U0067><U0069><U0076><U0068><U0065><U006C><U0061>"
+ 
+% abmon - The abbreviations for the months 
+% - Pha, Luh, Fam, Lam, Shu, Lwi, Lwa, Ngu, Khu, Tsh, L^ar, Nye
+abmon       "<U0050><U0068><U0061>";"<U004C><U0075><U0068>";/
+            "<U0046><U0061><U006D>";"<U004C><U0061><U006D>";/
+            "<U0053><U0068><U0075>";"<U004C><U0077><U0069>";/
+            "<U004C><U0077><U0061>";"<U004E><U0067><U0075>";/
+            "<U004B><U0068><U0075>";"<U0054><U0073><U0068>";/
+            "<U1E3C><U0061><U0072>";"<U004E><U0079><U0065>"
+ 
+% mon - The full names of the months - 
+% - Phando, Luhuhi, T^hafamuhwe, Lambamai, Shundunthule, Fulwi
+%   Fulwana, T^hangule, Khubvumedzi, Tshimedzi, L^ara, Nyendavhusiku
+mon         "<U0050><U0068><U0061><U006E><U0064><U006F>";/
+            "<U004C><U0075><U0068><U0075><U0068><U0069>";/
+            "<U1E70><U0068><U0061><U0066><U0061><U006D><U0075><U0068><U0077><U0065>";/
+            "<U004C><U0061><U006D><U0062><U0061><U006D><U0061><U0069>";/
+            "<U0053><U0068><U0075><U006E><U0064><U0075><U006E><U0074><U0068><U0075><U006C><U0065>";/
+            "<U0046><U0075><U006C><U0077><U0069>";/
+            "<U0046><U0075><U006C><U0077><U0061><U006E><U0061>";/
+            "<U1E70><U0068><U0061><U006E><U0067><U0075><U006C><U0065>";/
+            "<U004B><U0068><U0075><U0062><U0076><U0075><U006D><U0065><U0064><U007A><U0069>";/
+            "<U0054><U0073><U0068><U0069><U006D><U0065><U0064><U007A><U0069>";/
+            "<U1E3C><U0061><U0072><U0061>";/
+            "<U004E><U0079><U0065><U006E><U0064><U0061><U0076><U0068><U0075><U0073><U0069><U006B><U0075>"
+ 
+% Abreviated date and time representation to be referenced by the "%c" field descriptor -
+d_t_fmt     "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+%
+% "%a" (short weekday name),
+% "%d" (day of month as a decimal number),
+% "%b" (short month name),
+% "%Y" (year with century as a decimal number),
+% "%T" (24-hour clock time in format HH:MM:SS),
+% "%Z" (Time zone name)
+ 
+% Date representation to be referenced by the "%x" field descriptor -
+d_fmt   "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+% "%d/%m/%Y", day/month/year as decimal numbers (01/01/2000).
+ 
+% Time representation to be referenced by the "%X" field descriptor -
+t_fmt       "<U0025><U0054>"
+% "%T" (24-hour clock time in format HH:MM:SS)
+ 
+% Define representation of ante meridiem and post meridiem strings -
+am_pm       "";""
+% The "" mean 'default to "AM" and "PM".
+ 
+% Define time representation in 12-hour format with "am_pm", to be referenced by the "%r"
+t_fmt_ampm  ""
+% The "" means that this format is not supported.
+ 
+% Date representation not described in ISO/IEC 14652. Comes out as -
+% "%a %b %e %H:%M:%S %Z %Y" which is default "date" command output
+date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
+<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U005A><U0020><U0025><U0059>"
+%
+% %a - abbreviated weekday name,
+% %b - abreviated month name,
+% %e - day of month as a decimal number with leading space (1 to 31),
+% %H - hour (24-hour clock) as a decimal number (00 to 23),
+% %M - minute as a decimal number (00 to 59),
+% %S - seconds as a decimal number (00 to 59),
+% %Z - time-zone name,
+% %Y - year with century as a decimal number,e.g. 2001.
+ 
+END LC_TIME
+
+LC_MESSAGES
+% FIXME: Check both of these
+% The affirmative response - ee
+% "^[yYeE]"
+yesexpr     "<U005E><U005B><U0079><U0059><U0065><U0045><U005D>"
+
+% The negative response - hai
+% "^[nNhH]"
+noexpr      "<U005E><U005B><U006E><U004E><U0068><U0048><U005D>"
+END LC_MESSAGES
+
+LC_PAPER
+copy "en_ZA"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "en_ZA"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "en_ZA"
+END LC_MEASUREMENT
+
+LC_NAME
+% Format for addressing a person.
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+% "%d%t%g%t%m%t%f"
+% "Salutation",
+% "Empty string, or <Space>",
+% "First given name",
+% "Empty string, or <Space>",
+% "Middle names",
+% "Empty string, or <Space>",
+% "Clan names"
+ 
+% FIXME - define all the following name_*
+% General salutation for any sex
+% name_gen    ""
+
+% Salutation for unmarried females - ""
+% name_miss   ""
+ 
+% Salutation for males - ""
+% name_mr     ""
+
+% Salutation for married females - ""
+% name_mrs    ""
+
+% Salutation valid for all females - "" (no term)
+% name_ms     ""
+
+END LC_NAME
+
+LC_ADDRESS
+% Country name in Venda - "Afurika Tshipembe"
+country_name "<U0041><U0066><U0075><U0072><U0069><U006B><U0061><U0020>/
+<U0054><U0073><U0068><U0069><U0070><U0065><U006D><U0062><U0065>"
+ 
+% Abbreviated country postal name - "ZA"
+country_post "<U005A><U0041>"
+ 
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% FIXME define the following correctly
+% country_isbn ""
+ 
+% Language name in Venda - "Tshivend^a"
+lang_name "<U0054><U0073><U0068><U0069><U0076><U0065><U006E><U1E13><U0061>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+% "ve", "ven", :ven"
+lang_ab         "<U0076><U0065>"
+lang_term       "<U0076><U0065><U006E>"
+% FIXME there seems to be an error with the following but am unable to
+% determine what as it works correctly in other locale files
+% lang_lib        "<U0076><U0065><U006E>"
+ 
+% Representation of postal addresses (minus the addressee's name) in South
+% Africa. (Ignored for now)
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N", which gives -
+% "firm name",
+% "end of line",
+% "C/O address",
+% "end of line",
+% "department name",
+% "Building name",
+% "end of line",
+% "street or block name",
+% "space",
+% "house number or designation",
+% "space",
+% "floor number",
+% "space",
+% "room number, door designation",
+% "end of line",
+% "postal code",
+% "space",
+% "town, city",
+% "end of line",
+% "country designation for the <country_post> keyword",
+% "end of line
+%
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
+END LC_ADDRESS
+
diff -uNr libc.orig/localedata/locales/xh_ZA libc/localedata/locales/xh_ZA
--- libc.orig/localedata/locales/xh_ZA	2003-12-05 01:26:01.000000000 +0100
+++ libc/localedata/locales/xh_ZA	2004-11-08 18:03:04.000000000 +0100
@@ -2,21 +2,34 @@
 comment_char  %
 
 % ChangeLog
+% 0.2 (2004-11-02):
+%     2004-11-01 Dwayne Bailey <dwayne@translate.org.za>
+%       - Change all contact information
+%       - d_t_fmt, date_fmt cahnged day number %e to %-e to remove
+%         space padding in day.
+%       - Remove .* from yes/noexpr
+%     2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%       - Added country_ab2/3, country_num
+%       - Added country_car
+%     2004-03-29 Dwayne Bailey <dwayne@translate.org.za>
+%       - Added lang_ab and lang_term
+%     2004-02-27 Dwayne Bailey <dwayne@translate.org.za>
+%       - Add lang_name
 % 0.1 (2003-05-08):
-%   - Initial Xhosa locale by Zuza Software Foundation
+%    - Initial Xhosa locale by Zuza Software Foundation
 
 LC_IDENTIFICATION
 title      "Xhosa locale for South Africa"
-source     "Zuza Software Foundation"
-address    "PO Box 13412, Mowbray, 7705, South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "PO Box 28364, Sunnyside, 0132, South Africa"
 contact    "Dwayne Bailey"
 email      "dwayne@translate.org.za"
-tel        "+27 (0)21 448 9265"
-fax        "+27 (0)21 448 9574"
+tel        "+27 12 343 8389"
+fax        "+27 12 343 0389"
 language   "Xhosa"
 territory  "South Africa"
-revision   "0.1"
-date       "2003-05-08"
+revision   "0.2"
+date       "2004-11-02"
 %
 category  "xh_ZA:2003";LC_IDENTIFICATION
 category  "xh_ZA:2003";LC_CTYPE
@@ -99,9 +112,9 @@
             "<U0044><U0069><U0073><U0065><U006D><U0062><U0061>"
 %
 % Abreviated date and time representation to be referenced by the "%c" field descriptor -
-d_t_fmt     "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+d_t_fmt     "<U0025><U0061><U0020><U0025><U002D><U0065><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
 % "%a" (weekday name),
-% "%d" (day of month as a decimal number),
+% "%-e" (day of month as a decimal number),
 % "%b" (month name),
 % "%Y" (year with century as a decimal number),
 % "%T" (24-hour clock time in format HH:MM:SS),
@@ -125,8 +138,8 @@
 t_fmt_ampm  ""
 %
 % Date representation not described in ISO/IEC 14652. 
-% Comes out as - "%a %b %e %H:%M:%S %Z %Y" 
-date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
+% Comes out as - "%a %b %-e %H:%M:%S %Z %Y" 
+date_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U002D><U0065>/
 <U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
 <U0025><U005A><U0020><U0025><U0059>"
 % which is:
@@ -143,12 +156,12 @@
 
 LC_MESSAGES
 % The affirmative response -
-% "^[yYeE].*"
-yesexpr     "<U005E><U005B><U0079><U0059><U0065><U0045><U005D><U002E><U002A>"
+% "^[yYeE]"
+yesexpr     "<U005E><U005B><U0079><U0059><U0065><U0045><U005D>"
 %
 % The negative response -
-% "^[nNhH].*"
-noexpr      "<U005E><U005B><U006E><U004E><U0068><U0048><U005D><U002E><U002A>"
+% "^[nNhH]"
+noexpr      "<U005E><U005B><U006E><U004E><U0068><U0048><U005D>"
 %
 END LC_MESSAGES
 
@@ -198,9 +211,19 @@
 % Abbreviated country postal name - "ZA"
 country_post "<U005A><U0041>"
 %
-% ISO 3166 two letter country abbreviation - "ZA"
-country_ab2 "<U005A><U0041>"
-%
+% Language name in Sotho - "IsiXhosa"
+lang_name "<U0049><U0073><U0069><U0058><U0068><U006F><U0073><U0061>"
+
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+lang_ab   "<U0078><U0068>"
+lang_term "<U0078><U0068><U006F>"
+
 % Representation of postal addresses (minus the addressee's name) in South
 % Africa. (Ignored for now)
 postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
@@ -231,7 +254,13 @@
 % "country designation for the <country_post> keyword",
 % "end of line
 %
-country_ab3 "<U005A><U0041><U0046>"
-country_num 710
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
 END LC_ADDRESS
 
diff -uNr libc.orig/localedata/locales/zu_ZA libc/localedata/locales/zu_ZA
--- libc.orig/localedata/locales/zu_ZA	2003-12-05 01:26:01.000000000 +0100
+++ libc/localedata/locales/zu_ZA	2004-10-28 16:55:05.000000000 +0200
@@ -2,21 +2,37 @@
 comment_char  %
 
 % ChangeLog
+% 0.2 (2004-10-28):
+%     2004-10-28 Dwayne Bailey <dwayne@translate.org.za>
+%        - Added lang_lib
+%        - Reverted version number to 0.2 - needs more testing
+%     2004-10-27 Dwayne Bailey <dwayne@translate.org.za>
+%        - Add Translate.org.za to "source" for clarity
+%        - Update all contact details
+%        - Remove .* from yes/noexpr
+%     2004-03-30 Dwayne Bailey <dwayne@translate.org.za>
+%        - Added country_ab2/3, country_num
+%        - Added country_car
+%     2004-03-29 Dwayne Bailey <dwayne@translate.org.za>
+%       - Added lang_ab and lang_term
+%     2004-02-27 Dwayne Bailey <dwayne@translate.org.za>
+%	    - Correct capatilisation of lang_name
 % 0.1 (2003-04-19):
-%   - Initial Zulu locale by Zuza Software Foundation
+%     2003-04-19:
+%       - Initial Zulu locale by Zuza Software Foundation
 
 LC_IDENTIFICATION
 title      "Zulu locale for South Africa"
-source     "Zuza Software Foundation"
-address    "PO Box 13412, Mowbray, 7705, South Africa"
+source     "Zuza Software Foundation (Translate.org.za)"
+address    "Box 28364, Sunnyside, 0132, South Africa"
 contact    "Dwayne Bailey"
 email      "dwayne@translate.org.za"
-tel        "+27 (0)21 448 9265"
-fax        "+27 (0)21 448 9574"
+tel        "+27 12 343 0389"
+fax        "+27 12 448 0389"
 language   "Zulu"
 territory  "South Africa"
-revision   "0.1"
-date       "2003-04-19"
+revision   "0.2"
+date       "2004-10-28"
 %
 category  "zu_ZA:2003";LC_IDENTIFICATION
 category  "zu_ZA:2003";LC_CTYPE
@@ -145,13 +161,13 @@
 END LC_TIME
 
 LC_MESSAGES
-yesexpr     "<U005E><U005B><U0079><U0059><U005D><U002E><U002A>"
+yesexpr     "<U005E><U005B><U0079><U0059><U005D>"
 % The affirmative response -
-% "^[yY].*"
+% "^[yY]"
 %
-noexpr      "<U005E><U005B><U006E><U004E><U0063><U0043><U005D><U002E><U002A>"
+noexpr      "<U005E><U005B><U006E><U004E><U0063><U0043><U005D>"
 % The negative response -
-% "^[nNcC].*"
+% "^[nNcC]"
 %
 END LC_MESSAGES
 
@@ -201,12 +217,21 @@
 country_post "<U005A><U0041>"
 % Abbreviated country postal name - "ZA"
 %
-country_ab2 "<U005A><U0041>"
-% ISO 3166 two letter country abbreviation - "ZA"
-%
-lang_name "<U0049><U0073><U0069><U007A><U0075><U006C><U0075>"
-% Language name in Zulu - "Isizulu"
-%
+lang_name "<U0049><U0073><U0069><U005A><U0075><U006C><U0075>"
+% Language name in Zulu - "IsiZulu"
+
+% UN Geneve 1949:68 Distinguishing signs of vehicles in international traffic
+% http://www.unece.org/trans/conventn/disting-signs-5-2001.pdf
+% "ZA"
+country_car   "<U005A><U0041>"
+
+% ISO 639 two and three letter language names
+% see http://www.loc.gov/standards/iso639-2/englangn.html
+% "zu", "zul", "zul"
+lang_ab   "<U007A><U0075>"
+lang_term "<U007A><U0075><U006C>"
+lang_lib  "<U007A><U0075><U006C>"
+
 postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
 <U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
 <U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
@@ -237,7 +262,13 @@
 % "country designation for the <country_post> keyword",
 % "end of line
 %
-country_ab3 "<U005A><U0041><U0046>"
-country_num 710
+
+% ISO 3166 country number and 2 and 3 letter abreviations
+% http://www.unicode.org/onlinedat/countries.html
+% "ZA", "ZAF"
+country_num   710
+country_ab2   "<U005A><U0041>"
+country_ab3   "<U005A><U0041><U0046>"
+
 END LC_ADDRESS
 
diff -uNr libc.orig/localedata/SUPPORTED libc/localedata/SUPPORTED
--- libc.orig/localedata/SUPPORTED	2004-11-10 00:39:10.000000000 +0100
+++ libc/localedata/SUPPORTED	2004-11-10 01:28:15.000000000 +0100
@@ -276,6 +276,10 @@
 nl_NL@euro/ISO-8859-15 \
 nn_NO.UTF-8/UTF-8 \
 nn_NO/ISO-8859-1 \
+nr_ZA.UTF-8/UTF-8 \
+nr_ZA/ISO-8859-1 \
+nso_ZA.UTF-8/UTF-8 \
+nso_ZA/ISO-8859-10 \
 oc_FR.UTF-8/UTF-8 \
 oc_FR/ISO-8859-1 \
 om_ET/UTF-8 \
@@ -324,6 +328,8 @@
 sr_CS/ISO-8859-5 \
 sr_CS@Latn/ISO-8859-2 \
 sr_CS@ije/ISO-8859-5 \
+ss_ZA.UTF-8/UTF-8 \
+ss_ZA/ISO-8859-1 \
 st_ZA.UTF-8/UTF-8 \
 st_ZA/ISO-8859-1 \
 sv_FI.UTF-8/UTF-8 \
@@ -343,10 +349,14 @@
 tig_ER/UTF-8 \
 tl_PH.UTF-8/UTF-8 \
 tl_PH/ISO-8859-1 \
+tn_ZA.UTF-8/UTF-8 \
+tn_ZA/ISO-8859-10 \
 tr_CY.UTF-8/UTF-8 \
 tr_CY/ISO-8859-9 \
 tr_TR.UTF-8/UTF-8 \
 tr_TR/ISO-8859-9 \
+ts_ZA.UTF-8/UTF-8 \
+ts_ZA/ISO-8859-1 \
 tt_RU.KOI8-K/KOI8-K \
 tt_RU.UTF-8/UTF-8 \
 tt_RU/TATAR-CYR \
@@ -356,6 +366,7 @@
 uz_UZ.UTF-8/UTF-8 \
 uz_UZ/ISO-8859-1 \
 uz_UZ@cyrillic/UTF-8 \
+ve_ZA/UTF-8 \
 vi_VN.TCVN/TCVN5712-1 \
 vi_VN/UTF-8 \
 wa_BE.UTF-8/UTF-8 \
