--- glibc-2.5/sysdeps/powerpc/fpu/e_sqrt.c~	2006-10-06 18:33:08.000000000 +0200
+++ glibc-2.5/sysdeps/powerpc/fpu/e_sqrt.c	2006-10-06 18:32:25.000000000 +0200
@@ -17,6 +17,10 @@
    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
    02111-1307 USA.  */
 
+/* make sure bits/mathinline.h is included */
+#ifndef __USE_EXTERN_INLINES
+# define __USE_EXTERN_INLINES
+#endif
 #include <math.h>
 #include <math_private.h>
 #include <fenv_libc.h>
--- glibc-2.5/sysdeps/powerpc/fpu/e_sqrtf.c~	2006-04-14 07:44:30.000000000 +0200
+++ glibc-2.5/sysdeps/powerpc/fpu/e_sqrtf.c	2006-10-06 18:32:45.000000000 +0200
@@ -17,6 +17,10 @@
    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
    02111-1307 USA.  */
 
+/* make sure bits/mathinline.h is included */
+#ifndef __USE_EXTERN_INLINES
+# define __USE_EXTERN_INLINES
+#endif
 #include <math.h>
 #include <math_private.h>
 #include <fenv_libc.h>
