diff -ur glibc-2.3.6/include/netdb.h glibc-2.3.6-no-nss_hidden/include/netdb.h
--- glibc-2.3.6/include/netdb.h	2004-10-24 11:04:34.000000000 +0200
+++ glibc-2.3.6-no-nss_hidden/include/netdb.h	2005-12-05 19:46:11.077032520 +0100
@@ -202,10 +202,10 @@
 						struct __netgrent *result,
 						char *buffer, size_t buflen,
 						int *errnop);
-libnss_files_hidden_proto (_nss_files_parse_protoent)
-libnss_files_hidden_proto (_nss_files_parse_servent)
-libnss_files_hidden_proto (_nss_files_parse_netent)
-libnss_files_hidden_proto (_nss_netgroup_parseline)
+/*libnss_files_hidden_proto (_nss_files_parse_protoent)*/
+/*libnss_files_hidden_proto (_nss_files_parse_servent)*/
+/*libnss_files_hidden_proto (_nss_files_parse_netent)*/
+/*libnss_files_hidden_proto (_nss_netgroup_parseline)*/
 
 #define DECLARE_NSS_PROTOTYPES(service)					      \
 extern enum nss_status _nss_ ## service ## _setprotoent (int);		      \
diff -ur glibc-2.3.6/include/netinet/ether.h glibc-2.3.6-no-nss_hidden/include/netinet/ether.h
--- glibc-2.3.6/include/netinet/ether.h	2004-10-24 11:04:33.000000000 +0200
+++ glibc-2.3.6-no-nss_hidden/include/netinet/ether.h	2005-12-05 19:45:14.029705032 +0100
@@ -17,7 +17,7 @@
 extern int _nss_files_parse_etherent (char *line, struct etherent *result,
 				      struct parser_data *data,
 				      size_t datalen, int *errnop);
-libnss_files_hidden_proto (_nss_files_parse_etherent)
+/*libnss_files_hidden_proto (_nss_files_parse_etherent)*/
 
 #define DECLARE_NSS_PROTOTYPES(service)					      \
 extern enum nss_status _nss_ ## service ## _setetherent (int __stayopen);     \
diff -ur glibc-2.3.6/include/rpc/netdb.h glibc-2.3.6-no-nss_hidden/include/rpc/netdb.h
--- glibc-2.3.6/include/rpc/netdb.h	2004-10-24 11:04:33.000000000 +0200
+++ glibc-2.3.6-no-nss_hidden/include/rpc/netdb.h	2005-12-05 19:45:42.913314056 +0100
@@ -26,7 +26,7 @@
 extern int _nss_files_parse_rpcent (char *line, struct rpcent *result,
 				    struct parser_data *data,
 				    size_t datalen, int *errnop);
-libnss_files_hidden_proto (_nss_files_parse_rpcent)
+/*libnss_files_hidden_proto (_nss_files_parse_rpcent)*/
 
 #define DECLARE_NSS_PROTOTYPES(service)					      \
 extern enum nss_status _nss_ ## service ## _setrpcent (int);		      \
diff -ur glibc-2.3.6/nss/nss_files/files-netgrp.c glibc-2.3.6-no-nss_hidden/nss/nss_files/files-netgrp.c
--- glibc-2.3.6/nss/nss_files/files-netgrp.c	2004-10-24 11:05:12.000000000 +0200
+++ glibc-2.3.6-no-nss_hidden/nss/nss_files/files-netgrp.c	2005-12-05 19:59:34.097954784 +0100
@@ -274,7 +274,7 @@
 
   return status;
 }
-libnss_files_hidden_def (_nss_netgroup_parseline)
+/*libnss_files_hidden_def (_nss_netgroup_parseline)*/
 
 
 enum nss_status
