--- glibc-2.3.3/linuxthreads/sysdeps/powerpc/tcb-offsets.sym.orig	2003-12-18 10:59:10.000000000 +0100
+++ glibc-2.3.3/linuxthreads/sysdeps/powerpc/tcb-offsets.sym	2004-01-10 22:27:18.332718240 +0100
@@ -3,7 +3,6 @@
 
 # ifdef USE_TLS
 
--- Abuse tls.h macros to derive offsets relative to the thread register.
 #  undef __thread_register
 #  define __thread_register	((void *) 0)
 #  define thread_offsetof(mem)	((void *) &THREAD_SELF->p_##mem - (void *) 0)
@@ -13,6 +12,6 @@
 #  define thread_offsetof(mem)	offsetof (tcbhead_t, mem)
 
 # endif
-
+--
 MULTIPLE_THREADS_OFFSET		thread_offsetof (multiple_threads)
 
