--- glibc-2.7/time/tzfile.c.orig	2007-11-01 11:28:53.717388999 +0000
+++ glibc-2.7/time/tzfile.c	2007-11-01 11:29:33.711287677 +0000
@@ -381,7 +381,7 @@
       else
 	tzspec[tzspec_len - 1] = '\0';
     }
-  else if (sizeof (time_t) == 4 && tzhead.tzh_version != '\0')
+  else if (sizeof (time_t) == 4 && tzhead.tzh_version[0] != '\0')
     {
       /* Get the TZ string.  */
       if (__builtin_expect (fread_unlocked ((void *) &tzhead, sizeof (tzhead),
