--- glibc-2.13/sysdeps/unix/sysv/linux/faccessat.c~	2011-02-06 19:53:12.355202896 +0100
+++ glibc-2.13/sysdeps/unix/sysv/linux/faccessat.c	2011-02-06 19:53:14.418536233 +0100
@@ -17,6 +17,11 @@
    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
    02111-1307 USA.  */
 
+#ifndef __USE_EXTERN_INLINES
+# define __USE_EXTERN_INLINES
+# include <sys/stat.h>
+# undef __USE_EXTERN_INLINES
+#endif
 #include <errno.h>
 #include <fcntl.h>
 #include <stddef.h>
