--- glibc-2.1.2/linuxthreads/man/pthread_create.man.orig	Tue Sep 21 16:12:25 1999
+++ glibc-2.1.2/linuxthreads/man/pthread_create.man	Tue Sep 21 16:15:04 1999
@@ -31,10 +31,8 @@
 .SH ERRORS
 .TP
 !EAGAIN!
-not enough system resources to create a process for the new thread.
-.TP
-!EAGAIN!
-more than !PTHREAD_THREADS_MAX! threads are already active.
+not enough system resources to create a process for the new thread,
+or more than !PTHREAD_THREADS_MAX! threads are already active.
 
 .SH AUTHOR
 Xavier Leroy <Xavier.Leroy@inria.fr>
