diff -durN glibc-2.3.2.orig/elf/soinit.c glibc-2.3.2/elf/soinit.c
--- glibc-2.3.2.orig/elf/soinit.c	2001-11-16 00:09:20.000000000 +0100
+++ glibc-2.3.2/elf/soinit.c	2004-12-18 11:21:59.521615024 +0100
@@ -25,7 +25,7 @@
 }
 
 #ifdef HAVE_DWARF2_UNWIND_INFO
-static char __EH_FRAME_BEGIN__[]
+const static char __EH_FRAME_BEGIN__[]
      __attribute__ ((section (".eh_frame")))
      = { };
 # ifdef HAVE_DWARF2_UNWIND_INFO_STATIC
