--- glibc-2.1.3/sysdeps/unix/sysv/linux/paths.h.wiget	Wed Aug 12 19:05:51 1998
+++ glibc-2.1.3/sysdeps/unix/sysv/linux/paths.h	Fri Feb 25 09:08:26 2000
@@ -52,12 +52,12 @@
 #define	_PATH_KMEM	"/dev/kmem"
 #define	_PATH_LASTLOG	"/var/log/lastlog"
 #define	_PATH_MAILDIR	"/var/mail"
-#define	_PATH_MAN	"/usr/man"
+#define	_PATH_MAN	"/usr/share/man"
 #define	_PATH_MEM	"/dev/mem"
 #define	_PATH_MNTTAB	"/etc/fstab"
 #define	_PATH_MOUNTED	"/etc/mtab"
 #define	_PATH_NOLOGIN	"/etc/nologin"
-#define	_PATH_PRESERVE	"/var/preserve"
+#define	_PATH_PRESERVE	"/var/lib"
 #define	_PATH_RWHODIR	"/var/spool/rwho"
 #define	_PATH_SENDMAIL	"/usr/sbin/sendmail"
 #define	_PATH_SHADOW	"/etc/shadow"
--- glibc-2.1.3/nscd/nscd.conf.wiget	Mon Oct  4 21:51:37 1999
+++ glibc-2.1.3/nscd/nscd.conf	Fri Feb 25 09:10:01 2000
@@ -19,8 +19,8 @@
 #
 
 
-#	logfile			/var/log/nscd.log
-#	threads			6
+	logfile			/var/log/nscd
+	threads			6
 
 	debug-level		0
 
