--- glibc-2.3.2/localedata/SUPPORTED.orig	2003-06-26 09:29:43.000000000 +0200
+++ glibc-2.3.2/localedata/SUPPORTED	2003-09-08 15:46:27.593972416 +0200
@@ -25,6 +25,7 @@
 be_BY/CP1251 \
 bg_BG/CP1251 \
 br_FR/ISO-8859-1 \
+br_FR@euro/ISO-8859-15 \
 bs_BA/ISO-8859-2 \
 ca_ES/ISO-8859-1 \
 ca_ES@euro/ISO-8859-15 \
@@ -113,6 +114,7 @@
 ko_KR.EUC-KR/EUC-KR \
 ko_KR.UTF-8/UTF-8 \
 kw_GB/ISO-8859-1 \
+lg_UG/ISO-8859-10 \
 lo_LA/UTF-8 \
 lt_LT/ISO-8859-13 \
 lv_LV/ISO-8859-13 \
