diff -urN glibc-2.2.3.org/sysdeps/generic/bits/confname.h glibc-2.2.3/sysdeps/generic/bits/confname.h
--- glibc-2.2.3.org/sysdeps/generic/bits/confname.h	Tue May  8 23:51:52 2001
+++ glibc-2.2.3/sysdeps/generic/bits/confname.h	Tue May  8 23:52:08 2001
@@ -560,7 +560,7 @@
 #  define _CS_POSIX_V6_LPBIG_OFFBIG_LINTFLAGS _CS_POSIX_V6_LPBIG_OFFBIG_LINTFLAGS
 # endif
 
-    _CS_V6_WIDTH_RESTRICTED_ENVS,
+    _CS_V6_WIDTH_RESTRICTED_ENVS
 # define _CS_V6_WIDTH_RESTRICTED_ENVS	_CS_V6_WIDTH_RESTRICTED_ENVS
   };
 #endif
