diff -urN glibc-2.3.3.org/sysdeps/unix/sysv/linux/sys/sysctl.h glibc-2.3.3/sysdeps/unix/sysv/linux/sys/sysctl.h
--- glibc-2.3.3.org/sysdeps/unix/sysv/linux/sys/sysctl.h	2004-01-01 23:02:57.135529536 +0100
+++ glibc-2.3.3/sysdeps/unix/sysv/linux/sys/sysctl.h	2004-01-01 23:03:08.005270785 +0100
@@ -26,9 +26,6 @@
 #define _LINUX_KERNEL_H	1
 #define _LINUX_TYPES_H	1
 #define _LINUX_LIST_H	1
-/* We do need this one for the declarations in <linux/sysctl.h>,
-   since we've elided the inclusion of <linux/kernel.h> that gets them.  */
-#include <linux/compiler.h>
 #include <linux/sysctl.h>
 
 __BEGIN_DECLS
