diff -urN glibc-2.2.3.org/sysdeps/unix/sysv/linux/alpha/bits/termios.h glibc-2.2.3/sysdeps/unix/sysv/linux/alpha/bits/termios.h
--- glibc-2.2.3.org/sysdeps/unix/sysv/linux/alpha/bits/termios.h	Thu Jun 28 14:44:51 2001
+++ glibc-2.2.3/sysdeps/unix/sysv/linux/alpha/bits/termios.h	Thu Jun 28 14:45:07 2001
@@ -150,6 +150,8 @@
 #define  B3500000 00035
 #define  B4000000 00036
 
+#define __MAX_BAUD B4000000
+
 #define CSIZE	00001400
 #define   CS5	00000000
 #define   CS6	00000400
diff -urN glibc-2.2.3.org/sysdeps/unix/sysv/linux/bits/termios.h glibc-2.2.3/sysdeps/unix/sysv/linux/bits/termios.h
--- glibc-2.2.3.org/sysdeps/unix/sysv/linux/bits/termios.h	Thu Jun 28 14:44:52 2001
+++ glibc-2.2.3/sysdeps/unix/sysv/linux/bits/termios.h	Thu Jun 28 14:45:07 2001
@@ -165,6 +165,7 @@
 #define  B3000000 0010015
 #define  B3500000 0010016
 #define  B4000000 0010017
+#define __MAX_BAUD B4000000
 #ifdef __USE_MISC
 # define CIBAUD	  002003600000		/* input baud rate (not used) */
 # define CRTSCTS  020000000000		/* flow control */
diff -urN glibc-2.2.3.org/sysdeps/unix/sysv/linux/mips/bits/termios.h glibc-2.2.3/sysdeps/unix/sysv/linux/mips/bits/termios.h
--- glibc-2.2.3.org/sysdeps/unix/sysv/linux/mips/bits/termios.h	Thu Jun 28 14:44:55 2001
+++ glibc-2.2.3/sysdeps/unix/sysv/linux/mips/bits/termios.h	Thu Jun 28 14:45:07 2001
@@ -214,6 +214,7 @@
 # define  B3000000 0010015
 # define  B3500000 0010016
 # define  B4000000 0010017
+# define  __MAX_BAUD B4000000
 # define CIBAUD	  002003600000	/* input baud rate (not used) */
 # define CRTSCTS  020000000000		/* flow control */
 #endif
diff -urN glibc-2.2.3.org/sysdeps/unix/sysv/linux/powerpc/bits/termios.h glibc-2.2.3/sysdeps/unix/sysv/linux/powerpc/bits/termios.h
--- glibc-2.2.3.org/sysdeps/unix/sysv/linux/powerpc/bits/termios.h	Thu Jun 28 14:44:55 2001
+++ glibc-2.2.3/sysdeps/unix/sysv/linux/powerpc/bits/termios.h	Thu Jun 28 14:45:07 2001
@@ -154,6 +154,7 @@
 #define  B3000000 00034
 #define  B3500000 00035
 #define  B4000000 00036
+#define __MAX_BAUD B4000000
 
 #define CSIZE	00001400
 #define   CS5	00000000
diff -urN glibc-2.2.3.org/sysdeps/unix/sysv/linux/sparc/bits/termios.h glibc-2.2.3/sysdeps/unix/sysv/linux/sparc/bits/termios.h
--- glibc-2.2.3.org/sysdeps/unix/sysv/linux/sparc/bits/termios.h	Thu Jun 28 14:44:57 2001
+++ glibc-2.2.3/sysdeps/unix/sysv/linux/sparc/bits/termios.h	Thu Jun 28 14:45:07 2001
@@ -156,6 +156,7 @@
 #define B1152000 0x0000100d
 #define B1500000 0x0000100e
 #define B2000000 0x0000100f
+#define __MAX_BAUD B2000000
 
 #define CIBAUD	0x100f0000	/* input baud rate (not used) */
 #define CMSPAR	0x40000000	/* mark or space (stick) parity */
diff -urN glibc-2.2.3.org/sysdeps/unix/sysv/linux/speed.c glibc-2.2.3/sysdeps/unix/sysv/linux/speed.c
--- glibc-2.2.3.org/sysdeps/unix/sysv/linux/speed.c	Thu Jun 28 14:44:58 2001
+++ glibc-2.2.3/sysdeps/unix/sysv/linux/speed.c	Thu Jun 28 14:45:54 2001
@@ -60,7 +60,7 @@
      speed_t speed;
 {
   if ((speed & ~CBAUD) != 0
-      && (speed < B57600 || speed > B460800))
+      && (speed < B57600 || speed > __MAX_BAUD))
     {
       __set_errno (EINVAL);
       return -1;
@@ -82,7 +82,7 @@
      speed_t speed;
 {
   if ((speed & ~CBAUD) != 0
-      && (speed < B57600 || speed > B460800))
+      && (speed < B57600 || speed > __MAX_BAUD))
     {
       __set_errno (EINVAL);
       return -1;
